{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "from IPython.core.interactiveshell import InteractiveShell\n",
    "InteractiveShell.ast_node_interactivity = \"all\"\n",
    "\n",
    "import pandas as pd\n",
    "import csv\n",
    "import janitor\n",
    "import numpy as np\n",
    "import os\n",
    "import glob"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# read in original data\n",
    "\n",
    "path = '/Users/danielle/Documents/thesis/paper-abundance-tables/profiles/*'                     \n",
    "all_files = [name for name in glob.glob(path)]  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_from_each_file = []\n",
    "unclassified_list = []\n",
    "\n",
    "for f in all_files:\n",
    "    df = pd.read_csv(f, sep = '\\t', skiprows=[0,1,2], usecols = ['#clade_name','relative_abundance']) # read in dataframe \n",
    "    \n",
    "    id_1 = f.split('profiles/')[1] # add sample id from filename \n",
    "    id_2 = id_1.split('_S')[0]\n",
    "    df[\"sampleid\"] = id_2\n",
    "    \n",
    "    df.rename(columns = {'#clade_name':'taxa', 'relative_abundance':'abundance'}, inplace = True) \n",
    "    \n",
    "    df = df[df['taxa'].str.contains(\"\\|s__\")] # only keep genera\n",
    "    df = df[~df['taxa'].str.contains(\"\\|t__\")] \n",
    "    df[\"taxa\"] = df['taxa'].str.split(\"\\|g__\").str[-1]\n",
    "    df[\"taxa\"] = df['taxa'].str.split(\"\\|s__\").str[0]\n",
    "    \n",
    "    # percent unclassified organisms profiled with mgx\n",
    "    unclassified = sum(df.taxa.str.contains(\"_unclassified| noname | Candidatus\"))/df.shape[0]\n",
    "    unclassified_list.append(unclassified)\n",
    "    \n",
    "    # remove unclassified\n",
    "    df = df[~df.taxa.str.contains(\"_unclassified\")]\n",
    "    df = df[~df.taxa.str.contains(\"_noname\")]\n",
    "    df = df[~df.taxa.str.contains(\"Candidatus\")]\n",
    "    \n",
    "    # combine together taxa of the same genera\n",
    "    df = df.groupby(['taxa', 'sampleid'])['abundance'].sum().reset_index()    \n",
    "    \n",
    "    # convert to relative abundance\n",
    "    df[\"abundance\"] = df[\"abundance\"]/100.0\n",
    "    \n",
    "    df_from_each_file.append(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2.85990774818032"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# averaging mean unclassified across all dataframes\n",
    "np.mean(unclassified_list)*100"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "mgx = pd.concat(df_from_each_file, ignore_index=True) # concat all dataframes together"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>taxa</th>\n",
       "      <th>sampleid</th>\n",
       "      <th>abundance</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Agathobaculum</td>\n",
       "      <td>C0047-7E-1A</td>\n",
       "      <td>0.001349</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>Alistipes</td>\n",
       "      <td>C0047-7E-1A</td>\n",
       "      <td>0.036088</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>Anaerostipes</td>\n",
       "      <td>C0047-7E-1A</td>\n",
       "      <td>0.010047</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>Asaccharobacter</td>\n",
       "      <td>C0047-7E-1A</td>\n",
       "      <td>0.000012</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>Bacteroides</td>\n",
       "      <td>C0047-7E-1A</td>\n",
       "      <td>0.652440</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              taxa     sampleid  abundance\n",
       "0    Agathobaculum  C0047-7E-1A   0.001349\n",
       "1        Alistipes  C0047-7E-1A   0.036088\n",
       "2     Anaerostipes  C0047-7E-1A   0.010047\n",
       "3  Asaccharobacter  C0047-7E-1A   0.000012\n",
       "4      Bacteroides  C0047-7E-1A   0.652440"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mgx.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "mgx_reshaped = mgx.pivot_table(index = \"sampleid\", values=\"abundance\", columns = \"taxa\") # pivot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "mgx_reshaped.columns.name = None"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "mgx_reshaped = mgx_reshaped.rename_axis(None, axis=1).reset_index()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "mgx_reshaped = mgx_reshaped.fillna(0)  # fill in missing values with 0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sampleid</th>\n",
       "      <th>Absiella</th>\n",
       "      <th>Acetobacter</th>\n",
       "      <th>Acidaminococcus</th>\n",
       "      <th>Acinetobacter</th>\n",
       "      <th>Actinomyces</th>\n",
       "      <th>Actinotignum</th>\n",
       "      <th>Adlercreutzia</th>\n",
       "      <th>Aeriscardovia</th>\n",
       "      <th>Aeromonas</th>\n",
       "      <th>...</th>\n",
       "      <th>Stenotrophomonas</th>\n",
       "      <th>Streptococcus</th>\n",
       "      <th>Sutterella</th>\n",
       "      <th>Terrisporobacter</th>\n",
       "      <th>Turicibacter</th>\n",
       "      <th>Turicimonas</th>\n",
       "      <th>Tyzzerella</th>\n",
       "      <th>Varibaculum</th>\n",
       "      <th>Veillonella</th>\n",
       "      <th>Victivallis</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>C0047-7E-1A</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000156</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>C0052-7E-1A</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.001675</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.002813</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>C0053-8E-1A</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.001754</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000623</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.008166</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000670</td>\n",
       "      <td>0.000062</td>\n",
       "      <td>0.001062</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>C0058-4E-1A</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.003549</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.002141</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.001234</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>C0059-4E-1A</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.005600</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>125</td>\n",
       "      <td>C1177-4F-1A</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000079</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.009370</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>126</td>\n",
       "      <td>C1219-3F-1A</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000153</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.032457</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.025505</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>127</td>\n",
       "      <td>C2002-3E-1A</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.093448</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.001969</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000182</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.037591</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>128</td>\n",
       "      <td>C2014-2E-1A</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.002798</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.001256</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.013814</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>129</td>\n",
       "      <td>C2018-3E-1A</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.001675</td>\n",
       "      <td>0.000538</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.001675</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>130 rows × 140 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        sampleid  Absiella  Acetobacter  Acidaminococcus  Acinetobacter  \\\n",
       "0    C0047-7E-1A       0.0          0.0              0.0            0.0   \n",
       "1    C0052-7E-1A       0.0          0.0              0.0            0.0   \n",
       "2    C0053-8E-1A       0.0          0.0              0.0            0.0   \n",
       "3    C0058-4E-1A       0.0          0.0              0.0            0.0   \n",
       "4    C0059-4E-1A       0.0          0.0              0.0            0.0   \n",
       "..           ...       ...          ...              ...            ...   \n",
       "125  C1177-4F-1A       0.0          0.0              0.0            0.0   \n",
       "126  C1219-3F-1A       0.0          0.0              0.0            0.0   \n",
       "127  C2002-3E-1A       0.0          0.0              0.0            0.0   \n",
       "128  C2014-2E-1A       0.0          0.0              0.0            0.0   \n",
       "129  C2018-3E-1A       0.0          0.0              0.0            0.0   \n",
       "\n",
       "     Actinomyces  Actinotignum  Adlercreutzia  Aeriscardovia  Aeromonas  ...  \\\n",
       "0       0.000000           0.0       0.000000       0.000000        0.0  ...   \n",
       "1       0.000000           0.0       0.000000       0.000000        0.0  ...   \n",
       "2       0.001754           0.0       0.000623       0.000000        0.0  ...   \n",
       "3       0.000000           0.0       0.000000       0.000000        0.0  ...   \n",
       "4       0.000000           0.0       0.000000       0.000000        0.0  ...   \n",
       "..           ...           ...            ...            ...        ...  ...   \n",
       "125     0.000000           0.0       0.000000       0.000079        0.0  ...   \n",
       "126     0.000153           0.0       0.000000       0.000000        0.0  ...   \n",
       "127     0.000000           0.0       0.000000       0.000000        0.0  ...   \n",
       "128     0.002798           0.0       0.000000       0.000000        0.0  ...   \n",
       "129     0.000000           0.0       0.000000       0.000000        0.0  ...   \n",
       "\n",
       "     Stenotrophomonas  Streptococcus  Sutterella  Terrisporobacter  \\\n",
       "0                 0.0       0.000000    0.000156               0.0   \n",
       "1                 0.0       0.001675    0.000000               0.0   \n",
       "2                 0.0       0.008166    0.000000               0.0   \n",
       "3                 0.0       0.003549    0.000000               0.0   \n",
       "4                 0.0       0.005600    0.000000               0.0   \n",
       "..                ...            ...         ...               ...   \n",
       "125               0.0       0.009370    0.000000               0.0   \n",
       "126               0.0       0.032457    0.000000               0.0   \n",
       "127               0.0       0.093448    0.000000               0.0   \n",
       "128               0.0       0.001256    0.000000               0.0   \n",
       "129               0.0       0.001675    0.000538               0.0   \n",
       "\n",
       "     Turicibacter  Turicimonas  Tyzzerella  Varibaculum  Veillonella  \\\n",
       "0        0.000000     0.000000    0.000000          0.0     0.000000   \n",
       "1        0.002813     0.000000    0.000000          0.0     0.000000   \n",
       "2        0.000670     0.000062    0.001062          0.0     0.000000   \n",
       "3        0.002141     0.000000    0.000000          0.0     0.001234   \n",
       "4        0.000000     0.000000    0.000000          0.0     0.000000   \n",
       "..            ...          ...         ...          ...          ...   \n",
       "125      0.000000     0.000000    0.000000          0.0     0.000000   \n",
       "126      0.000000     0.000000    0.000000          0.0     0.025505   \n",
       "127      0.001969     0.000000    0.000182          0.0     0.037591   \n",
       "128      0.000000     0.000000    0.000000          0.0     0.013814   \n",
       "129      0.000000     0.000000    0.000000          0.0     0.001675   \n",
       "\n",
       "     Victivallis  \n",
       "0            0.0  \n",
       "1            0.0  \n",
       "2            0.0  \n",
       "3            0.0  \n",
       "4            0.0  \n",
       "..           ...  \n",
       "125          0.0  \n",
       "126          0.0  \n",
       "127          0.0  \n",
       "128          0.0  \n",
       "129          0.0  \n",
       "\n",
       "[130 rows x 140 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mgx_reshaped"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "# adding age metadata\n",
    "age = pd.read_csv(\"/Users/danielle/Documents/thesis/paper-abundance-tables/metadata/joined.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>subject</th>\n",
       "      <th>timepoint</th>\n",
       "      <th>batch</th>\n",
       "      <th>childGender</th>\n",
       "      <th>correctedAgeDays</th>\n",
       "      <th>mother_HHS</th>\n",
       "      <th>childBMI</th>\n",
       "      <th>hasScan</th>\n",
       "      <th>mcDespotProcessed</th>\n",
       "      <th>...</th>\n",
       "      <th>rt_Pallidum</th>\n",
       "      <th>rt_Hippocampus</th>\n",
       "      <th>rt_Amygdala</th>\n",
       "      <th>rt_Insula</th>\n",
       "      <th>rt_Operculum</th>\n",
       "      <th>hires_total</th>\n",
       "      <th>cerebellar</th>\n",
       "      <th>neocortical</th>\n",
       "      <th>limbic</th>\n",
       "      <th>subcortical</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>C0052_5F_1A</td>\n",
       "      <td>52</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>Male</td>\n",
       "      <td>2824.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>16.728348</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1315.875122</td>\n",
       "      <td>2491.843506</td>\n",
       "      <td>1107.906250</td>\n",
       "      <td>536.937500</td>\n",
       "      <td>570.96875</td>\n",
       "      <td>264479.536024</td>\n",
       "      <td>89286.656250</td>\n",
       "      <td>127223.941542</td>\n",
       "      <td>34246.781738</td>\n",
       "      <td>47968.938232</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>C0126_8F_1A</td>\n",
       "      <td>126</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>Male</td>\n",
       "      <td>2261.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1486.031250</td>\n",
       "      <td>2737.625000</td>\n",
       "      <td>1210.000122</td>\n",
       "      <td>680.625000</td>\n",
       "      <td>616.34375</td>\n",
       "      <td>320037.446868</td>\n",
       "      <td>100407.316406</td>\n",
       "      <td>163357.565277</td>\n",
       "      <td>41079.502685</td>\n",
       "      <td>56272.565185</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>C0126_8F_1B</td>\n",
       "      <td>126</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>Male</td>\n",
       "      <td>2261.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1486.031250</td>\n",
       "      <td>2737.625000</td>\n",
       "      <td>1210.000122</td>\n",
       "      <td>680.625000</td>\n",
       "      <td>616.34375</td>\n",
       "      <td>320037.446868</td>\n",
       "      <td>100407.316406</td>\n",
       "      <td>163357.565277</td>\n",
       "      <td>41079.502685</td>\n",
       "      <td>56272.565185</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>C0127_6F_1B</td>\n",
       "      <td>127</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>Female</td>\n",
       "      <td>3367.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1338.562500</td>\n",
       "      <td>2230.937500</td>\n",
       "      <td>1089.000000</td>\n",
       "      <td>654.156311</td>\n",
       "      <td>567.18750</td>\n",
       "      <td>270177.880217</td>\n",
       "      <td>86435.597656</td>\n",
       "      <td>135330.938994</td>\n",
       "      <td>34012.343781</td>\n",
       "      <td>48411.343567</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>C0286_5F_1A</td>\n",
       "      <td>286</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>Male</td>\n",
       "      <td>1926.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1603.250000</td>\n",
       "      <td>2457.812500</td>\n",
       "      <td>1070.093750</td>\n",
       "      <td>616.343750</td>\n",
       "      <td>548.28125</td>\n",
       "      <td>284418.066527</td>\n",
       "      <td>88447.218750</td>\n",
       "      <td>143566.502440</td>\n",
       "      <td>36700.813782</td>\n",
       "      <td>52404.345337</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 115 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        sample  subject  timepoint  batch childGender  correctedAgeDays  \\\n",
       "0  C0052_5F_1A       52          5      1        Male            2824.0   \n",
       "1  C0126_8F_1A      126          8      1        Male            2261.0   \n",
       "2  C0126_8F_1B      126          8      1        Male            2261.0   \n",
       "3  C0127_6F_1B      127          6      1      Female            3367.0   \n",
       "4  C0286_5F_1A      286          5      1        Male            1926.0   \n",
       "\n",
       "   mother_HHS   childBMI  hasScan  mcDespotProcessed  ...  rt_Pallidum  \\\n",
       "0         6.0  16.728348      1.0                1.0  ...  1315.875122   \n",
       "1         7.0        NaN      1.0                0.0  ...  1486.031250   \n",
       "2         7.0        NaN      1.0                0.0  ...  1486.031250   \n",
       "3         7.0        NaN      1.0                1.0  ...  1338.562500   \n",
       "4         6.0        NaN      1.0                0.0  ...  1603.250000   \n",
       "\n",
       "   rt_Hippocampus  rt_Amygdala   rt_Insula  rt_Operculum    hires_total  \\\n",
       "0     2491.843506  1107.906250  536.937500     570.96875  264479.536024   \n",
       "1     2737.625000  1210.000122  680.625000     616.34375  320037.446868   \n",
       "2     2737.625000  1210.000122  680.625000     616.34375  320037.446868   \n",
       "3     2230.937500  1089.000000  654.156311     567.18750  270177.880217   \n",
       "4     2457.812500  1070.093750  616.343750     548.28125  284418.066527   \n",
       "\n",
       "      cerebellar    neocortical        limbic   subcortical  \n",
       "0   89286.656250  127223.941542  34246.781738  47968.938232  \n",
       "1  100407.316406  163357.565277  41079.502685  56272.565185  \n",
       "2  100407.316406  163357.565277  41079.502685  56272.565185  \n",
       "3   86435.597656  135330.938994  34012.343781  48411.343567  \n",
       "4   88447.218750  143566.502440  36700.813782  52404.345337  \n",
       "\n",
       "[5 rows x 115 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "age.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "# change characters in sample names \n",
    "age[\"sample\"] = age[\"sample\"].str.replace(\"_\",'-')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "age[\"AgeMonths\"] = age[\"correctedAgeDays\"]/30.0\n",
    "age.rename(columns = {'sample':'sampleid'}, inplace = True) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "# make age dictionary\n",
    "agedict = {str(s): {} for s in age[\"sampleid\"]}\n",
    "for index, row in age.iterrows():\n",
    "    age_months = row[\"AgeMonths\"]\n",
    "    agedict[row[\"sampleid\"]]= age_months"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "mgx_reshaped[\"AgeMonths\"]= mgx_reshaped[\"sampleid\"].map(agedict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "def assign_dev_stage(x):\n",
    "    if x<15:\n",
    "        return \"less than 15 months\"\n",
    "    elif 15 <=x<= 30:\n",
    "        return \"15 to 30 months\"\n",
    "    elif x>30:\n",
    "        return \"older than 30 months\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "mgx_reshaped['dev_stage'] = mgx_reshaped['AgeMonths'].apply(assign_dev_stage)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "# re-order columns\n",
    "\n",
    "cols_to_order = ['sampleid','AgeMonths', 'dev_stage']\n",
    "new_columns = cols_to_order + (mgx_reshaped.columns.drop(cols_to_order).tolist())\n",
    "mgx_reshaped = mgx_reshaped[new_columns]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sampleid</th>\n",
       "      <th>AgeMonths</th>\n",
       "      <th>dev_stage</th>\n",
       "      <th>Absiella</th>\n",
       "      <th>Acetobacter</th>\n",
       "      <th>Acidaminococcus</th>\n",
       "      <th>Acinetobacter</th>\n",
       "      <th>Actinomyces</th>\n",
       "      <th>Actinotignum</th>\n",
       "      <th>Adlercreutzia</th>\n",
       "      <th>...</th>\n",
       "      <th>Stenotrophomonas</th>\n",
       "      <th>Streptococcus</th>\n",
       "      <th>Sutterella</th>\n",
       "      <th>Terrisporobacter</th>\n",
       "      <th>Turicibacter</th>\n",
       "      <th>Turicimonas</th>\n",
       "      <th>Tyzzerella</th>\n",
       "      <th>Varibaculum</th>\n",
       "      <th>Veillonella</th>\n",
       "      <th>Victivallis</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>C0047-7E-1A</td>\n",
       "      <td>97.233333</td>\n",
       "      <td>older than 30 months</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000156</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>C0052-7E-1A</td>\n",
       "      <td>127.333333</td>\n",
       "      <td>older than 30 months</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.001675</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.002813</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>C0053-8E-1A</td>\n",
       "      <td>105.700000</td>\n",
       "      <td>older than 30 months</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.001754</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000623</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.008166</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000670</td>\n",
       "      <td>0.000062</td>\n",
       "      <td>0.001062</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>C0058-4E-1A</td>\n",
       "      <td>137.766667</td>\n",
       "      <td>older than 30 months</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.003549</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.002141</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.001234</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>C0059-4E-1A</td>\n",
       "      <td>137.766667</td>\n",
       "      <td>older than 30 months</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.005600</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>125</td>\n",
       "      <td>C1177-4F-1A</td>\n",
       "      <td>1.666667</td>\n",
       "      <td>less than 15 months</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.009370</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>126</td>\n",
       "      <td>C1219-3F-1A</td>\n",
       "      <td>1.266667</td>\n",
       "      <td>less than 15 months</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000153</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.032457</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.025505</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>127</td>\n",
       "      <td>C2002-3E-1A</td>\n",
       "      <td>8.600000</td>\n",
       "      <td>less than 15 months</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.093448</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.001969</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000182</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.037591</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>128</td>\n",
       "      <td>C2014-2E-1A</td>\n",
       "      <td>6.100000</td>\n",
       "      <td>less than 15 months</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.002798</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.001256</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.013814</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>129</td>\n",
       "      <td>C2018-3E-1A</td>\n",
       "      <td>12.100000</td>\n",
       "      <td>less than 15 months</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.001675</td>\n",
       "      <td>0.000538</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.001675</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>130 rows × 142 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        sampleid   AgeMonths             dev_stage  Absiella  Acetobacter  \\\n",
       "0    C0047-7E-1A   97.233333  older than 30 months       0.0          0.0   \n",
       "1    C0052-7E-1A  127.333333  older than 30 months       0.0          0.0   \n",
       "2    C0053-8E-1A  105.700000  older than 30 months       0.0          0.0   \n",
       "3    C0058-4E-1A  137.766667  older than 30 months       0.0          0.0   \n",
       "4    C0059-4E-1A  137.766667  older than 30 months       0.0          0.0   \n",
       "..           ...         ...                   ...       ...          ...   \n",
       "125  C1177-4F-1A    1.666667   less than 15 months       0.0          0.0   \n",
       "126  C1219-3F-1A    1.266667   less than 15 months       0.0          0.0   \n",
       "127  C2002-3E-1A    8.600000   less than 15 months       0.0          0.0   \n",
       "128  C2014-2E-1A    6.100000   less than 15 months       0.0          0.0   \n",
       "129  C2018-3E-1A   12.100000   less than 15 months       0.0          0.0   \n",
       "\n",
       "     Acidaminococcus  Acinetobacter  Actinomyces  Actinotignum  Adlercreutzia  \\\n",
       "0                0.0            0.0     0.000000           0.0       0.000000   \n",
       "1                0.0            0.0     0.000000           0.0       0.000000   \n",
       "2                0.0            0.0     0.001754           0.0       0.000623   \n",
       "3                0.0            0.0     0.000000           0.0       0.000000   \n",
       "4                0.0            0.0     0.000000           0.0       0.000000   \n",
       "..               ...            ...          ...           ...            ...   \n",
       "125              0.0            0.0     0.000000           0.0       0.000000   \n",
       "126              0.0            0.0     0.000153           0.0       0.000000   \n",
       "127              0.0            0.0     0.000000           0.0       0.000000   \n",
       "128              0.0            0.0     0.002798           0.0       0.000000   \n",
       "129              0.0            0.0     0.000000           0.0       0.000000   \n",
       "\n",
       "     ...  Stenotrophomonas  Streptococcus  Sutterella  Terrisporobacter  \\\n",
       "0    ...               0.0       0.000000    0.000156               0.0   \n",
       "1    ...               0.0       0.001675    0.000000               0.0   \n",
       "2    ...               0.0       0.008166    0.000000               0.0   \n",
       "3    ...               0.0       0.003549    0.000000               0.0   \n",
       "4    ...               0.0       0.005600    0.000000               0.0   \n",
       "..   ...               ...            ...         ...               ...   \n",
       "125  ...               0.0       0.009370    0.000000               0.0   \n",
       "126  ...               0.0       0.032457    0.000000               0.0   \n",
       "127  ...               0.0       0.093448    0.000000               0.0   \n",
       "128  ...               0.0       0.001256    0.000000               0.0   \n",
       "129  ...               0.0       0.001675    0.000538               0.0   \n",
       "\n",
       "     Turicibacter  Turicimonas  Tyzzerella  Varibaculum  Veillonella  \\\n",
       "0        0.000000     0.000000    0.000000          0.0     0.000000   \n",
       "1        0.002813     0.000000    0.000000          0.0     0.000000   \n",
       "2        0.000670     0.000062    0.001062          0.0     0.000000   \n",
       "3        0.002141     0.000000    0.000000          0.0     0.001234   \n",
       "4        0.000000     0.000000    0.000000          0.0     0.000000   \n",
       "..            ...          ...         ...          ...          ...   \n",
       "125      0.000000     0.000000    0.000000          0.0     0.000000   \n",
       "126      0.000000     0.000000    0.000000          0.0     0.025505   \n",
       "127      0.001969     0.000000    0.000182          0.0     0.037591   \n",
       "128      0.000000     0.000000    0.000000          0.0     0.013814   \n",
       "129      0.000000     0.000000    0.000000          0.0     0.001675   \n",
       "\n",
       "     Victivallis  \n",
       "0            0.0  \n",
       "1            0.0  \n",
       "2            0.0  \n",
       "3            0.0  \n",
       "4            0.0  \n",
       "..           ...  \n",
       "125          0.0  \n",
       "126          0.0  \n",
       "127          0.0  \n",
       "128          0.0  \n",
       "129          0.0  \n",
       "\n",
       "[130 rows x 142 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mgx_reshaped"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "mgx_reshaped.to_csv('mgx_abund_df.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "# read in 16S data\n",
    "amplicon = pd.read_csv(\"/Users/danielle/Documents/thesis/paper-abundance-tables/dada2_output_exported/feature-table.txt\",\n",
    "                      skiprows=[0], sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "amplicon.drop(['#OTU ID'], axis = 1, inplace = True)\n",
    "amplicon.rename(columns = {\"taxonomy\": \"taxa\"}, inplace = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "cols = list(amplicon.columns)\n",
    "cols = [cols[-1]] + cols[:-1]\n",
    "amplicon = amplicon[cols]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "# clean taxonomic names\n",
    "# keep only genera\n",
    "amplicon[\"taxa\"] = amplicon['taxa'].str.split(\"D_5__\", expand=True)[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "amplicon[\"taxa\"] = amplicon['taxa'].str.split(\";\", expand=True)[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "amplicon[\"taxa\"] = amplicon['taxa'].str.split(\"_\", expand=True)[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "amplicon[\"taxa\"] = amplicon['taxa'].str.split(\"-\", expand=True)[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "amplicon[\"taxa\"] = amplicon[\"taxa\"].str.strip('[]').astype(str)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>taxa</th>\n",
       "      <th>C0047-7E-1A</th>\n",
       "      <th>C0052-7E-1A</th>\n",
       "      <th>C0053-8E-1A</th>\n",
       "      <th>C0058-4E-1A</th>\n",
       "      <th>C0059-4E-1A</th>\n",
       "      <th>C0106-6E-1A</th>\n",
       "      <th>C0107-4E-1A</th>\n",
       "      <th>C0123-3E-1A</th>\n",
       "      <th>C0126-9E-1A</th>\n",
       "      <th>...</th>\n",
       "      <th>C1102-1F-1A</th>\n",
       "      <th>C1109-4F-1A</th>\n",
       "      <th>C1115-4F-1A</th>\n",
       "      <th>C1135-4F-1A</th>\n",
       "      <th>C1160-3F-1A</th>\n",
       "      <th>C1177-4F-1A</th>\n",
       "      <th>C1219-3F-1A</th>\n",
       "      <th>C2002-3E-1A</th>\n",
       "      <th>C2014-2E-1A</th>\n",
       "      <th>C2018-3E-1A</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Bacteroides</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1163.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>214.0</td>\n",
       "      <td>1476.0</td>\n",
       "      <td>...</td>\n",
       "      <td>369.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>336.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>577.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>376.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>Escherichia</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>42.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>637.0</td>\n",
       "      <td>593.0</td>\n",
       "      <td>1559.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>14633.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>1574.0</td>\n",
       "      <td>13.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>Prevotella</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1595.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>826.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>8823.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>Bacteroides</td>\n",
       "      <td>0.0</td>\n",
       "      <td>114.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>79.0</td>\n",
       "      <td>533.0</td>\n",
       "      <td>...</td>\n",
       "      <td>420.0</td>\n",
       "      <td>19.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>32.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1801.0</td>\n",
       "      <td>889.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>Bacteroides</td>\n",
       "      <td>1289.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2493.0</td>\n",
       "      <td>5670.0</td>\n",
       "      <td>4056.0</td>\n",
       "      <td>1143.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>831.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>13995.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>198.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 131 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          taxa  C0047-7E-1A  C0052-7E-1A  C0053-8E-1A  C0058-4E-1A  \\\n",
       "0  Bacteroides          0.0       1163.0          0.0          0.0   \n",
       "1  Escherichia          0.0          5.0         42.0          0.0   \n",
       "2   Prevotella          0.0       1595.0          0.0          0.0   \n",
       "3  Bacteroides          0.0        114.0          0.0          0.0   \n",
       "4  Bacteroides       1289.0          0.0          0.0       2493.0   \n",
       "\n",
       "   C0059-4E-1A  C0106-6E-1A  C0107-4E-1A  C0123-3E-1A  C0126-9E-1A  ...  \\\n",
       "0          0.0          0.0          0.0        214.0       1476.0  ...   \n",
       "1          0.0          7.0          0.0          0.0          0.0  ...   \n",
       "2          0.0          0.0          7.0          0.0          0.0  ...   \n",
       "3          0.0          0.0          0.0         79.0        533.0  ...   \n",
       "4       5670.0       4056.0       1143.0          0.0          0.0  ...   \n",
       "\n",
       "   C1102-1F-1A  C1109-4F-1A  C1115-4F-1A  C1135-4F-1A  C1160-3F-1A  \\\n",
       "0        369.0          0.0          0.0          0.0          9.0   \n",
       "1          4.0          0.0        637.0        593.0       1559.0   \n",
       "2        826.0          0.0          0.0          0.0          0.0   \n",
       "3        420.0         19.0          0.0         32.0          0.0   \n",
       "4        831.0          0.0          0.0          0.0          0.0   \n",
       "\n",
       "   C1177-4F-1A  C1219-3F-1A  C2002-3E-1A  C2014-2E-1A  C2018-3E-1A  \n",
       "0        336.0          0.0        577.0          0.0        376.0  \n",
       "1          0.0      14633.0         18.0       1574.0         13.0  \n",
       "2          0.0          0.0          0.0          0.0       8823.0  \n",
       "3          0.0       1801.0        889.0          0.0          0.0  \n",
       "4      13995.0          0.0        198.0          0.0          0.0  \n",
       "\n",
       "[5 rows x 131 columns]"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "amplicon.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "# combine together taxa of the same genera\n",
    "amplicon = amplicon.groupby(['taxa']).sum().reset_index()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "amplicon.sort_values(by = \"taxa\", ascending=True, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "# calculate relative abundances\n",
    "bug_sum = amplicon.sum(axis = 0, skipna = True)\n",
    "amplicon.iloc[:, 1:] = amplicon.iloc[:, 1:]/bug_sum"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "amplicon_melt = pd.melt(amplicon, id_vars=[\"taxa\"], var_name = \"sampleid\", value_name = \"amplicon_abund\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "# remove if genus is less than 5 characters\n",
    "remove = list(set([name for name in amplicon_melt[\"taxa\"] if len(name)<5]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "# remove if numbers in genus\n",
    "remove2 = list(set([name for name in amplicon_melt[\"taxa\"] if not name.isalpha()]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "False"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\"Ruminococcaceae_UCG-010\".isalpha()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "# remove if Candidatus in genus\n",
    "remove3 = list(set([name for name in amplicon_melt[\"taxa\"] if \"Candidatus\" in name]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "remove_list = remove + remove2 + remove3 + [\"uncultured\", \"Family\", \"None\", \"Clade\", \"Subgroup\", \"Ruminococcaceae\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "9.239130434782608"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# percent unclassified organisms profiled with 16S\n",
    "np.sum([amplicon_melt.taxa.isin(remove_list)])/(amplicon_melt[\"taxa\"].shape[0])*100"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "amplicon_melt = amplicon_melt[~amplicon_melt.taxa.isin(remove_list)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "amplicon_reshaped = amplicon_melt.pivot_table(index = \"sampleid\", values=\"amplicon_abund\", columns = \"taxa\") # pivot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [],
   "source": [
    "amplicon_reshaped.columns.name = None"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "amplicon_reshaped = amplicon_reshaped.rename_axis(None, axis=1).reset_index()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [],
   "source": [
    "amplicon_reshaped = amplicon_reshaped.fillna(0)  # fill in missing values with 0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [],
   "source": [
    "amplicon_reshaped = amplicon_reshaped.loc[:,~amplicon_reshaped.columns.str.contains(r'[0-9]', case=False)] "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sampleid</th>\n",
       "      <th>Acetanaerobacterium</th>\n",
       "      <th>Achromobacter</th>\n",
       "      <th>Acidaminococcus</th>\n",
       "      <th>Acinetobacter</th>\n",
       "      <th>Actinobacillus</th>\n",
       "      <th>Actinomyces</th>\n",
       "      <th>Adlercreutzia</th>\n",
       "      <th>Aeromonas</th>\n",
       "      <th>Agathobacter</th>\n",
       "      <th>...</th>\n",
       "      <th>Sulfitobacter</th>\n",
       "      <th>Sutterella</th>\n",
       "      <th>Terrisporobacter</th>\n",
       "      <th>Turicibacter</th>\n",
       "      <th>Tyzzerella</th>\n",
       "      <th>Ulvibacter</th>\n",
       "      <th>Varibaculum</th>\n",
       "      <th>Veillonella</th>\n",
       "      <th>Victivallis</th>\n",
       "      <th>Virgulinella</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>C0047-7E-1A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.022783</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.018552</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000271</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>C0052-7E-1A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.013665</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.006443</td>\n",
       "      <td>0.005048</td>\n",
       "      <td>0.008125</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000657</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>C0053-8E-1A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000678</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.005422</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001355</td>\n",
       "      <td>0.030497</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000387</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>C0058-4E-1A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000273</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.011691</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002065</td>\n",
       "      <td>0.006937</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.002650</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>C0059-4E-1A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000403</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.101117</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.004730</td>\n",
       "      <td>0.002290</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.001359</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>125</td>\n",
       "      <td>C1177-4F-1A</td>\n",
       "      <td>0.000157</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.013042</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>126</td>\n",
       "      <td>C1219-3F-1A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000028</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.060177</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>127</td>\n",
       "      <td>C2002-3E-1A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.031022</td>\n",
       "      <td>0.005005</td>\n",
       "      <td>0.016784</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.017604</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>128</td>\n",
       "      <td>C2014-2E-1A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000480</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00012</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.026844</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>129</td>\n",
       "      <td>C2018-3E-1A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.020835</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.027287</td>\n",
       "      <td>0.000812</td>\n",
       "      <td>0.000143</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.002342</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>130 rows × 168 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        sampleid  Acetanaerobacterium  Achromobacter  Acidaminococcus  \\\n",
       "0    C0047-7E-1A             0.000000            0.0              0.0   \n",
       "1    C0052-7E-1A             0.000000            0.0              0.0   \n",
       "2    C0053-8E-1A             0.000000            0.0              0.0   \n",
       "3    C0058-4E-1A             0.000000            0.0              0.0   \n",
       "4    C0059-4E-1A             0.000000            0.0              0.0   \n",
       "..           ...                  ...            ...              ...   \n",
       "125  C1177-4F-1A             0.000157            0.0              0.0   \n",
       "126  C1219-3F-1A             0.000000            0.0              0.0   \n",
       "127  C2002-3E-1A             0.000000            0.0              0.0   \n",
       "128  C2014-2E-1A             0.000000            0.0              0.0   \n",
       "129  C2018-3E-1A             0.000000            0.0              0.0   \n",
       "\n",
       "     Acinetobacter  Actinobacillus  Actinomyces  Adlercreutzia  Aeromonas  \\\n",
       "0              0.0             0.0     0.000000       0.000000        0.0   \n",
       "1              0.0             0.0     0.000000       0.000000        0.0   \n",
       "2              0.0             0.0     0.000678       0.000000        0.0   \n",
       "3              0.0             0.0     0.000000       0.000273        0.0   \n",
       "4              0.0             0.0     0.000000       0.000403        0.0   \n",
       "..             ...             ...          ...            ...        ...   \n",
       "125            0.0             0.0     0.000000       0.000000        0.0   \n",
       "126            0.0             0.0     0.000028       0.000000        0.0   \n",
       "127            0.0             0.0     0.000000       0.000000        0.0   \n",
       "128            0.0             0.0     0.000480       0.000000        0.0   \n",
       "129            0.0             0.0     0.000000       0.000000        0.0   \n",
       "\n",
       "     Agathobacter  ...  Sulfitobacter  Sutterella  Terrisporobacter  \\\n",
       "0        0.022783  ...        0.00000    0.018552          0.000000   \n",
       "1        0.013665  ...        0.00000    0.006443          0.005048   \n",
       "2        0.005422  ...        0.00000    0.000000          0.000000   \n",
       "3        0.011691  ...        0.00000    0.000000          0.002065   \n",
       "4        0.101117  ...        0.00000    0.000000          0.004730   \n",
       "..            ...  ...            ...         ...               ...   \n",
       "125      0.000000  ...        0.00000    0.013042          0.000000   \n",
       "126      0.000000  ...        0.00000    0.000000          0.000000   \n",
       "127      0.000000  ...        0.00000    0.000000          0.031022   \n",
       "128      0.000000  ...        0.00012    0.000000          0.000000   \n",
       "129      0.020835  ...        0.00000    0.027287          0.000812   \n",
       "\n",
       "     Turicibacter  Tyzzerella  Ulvibacter  Varibaculum  Veillonella  \\\n",
       "0        0.000271    0.000000         0.0          0.0     0.000000   \n",
       "1        0.008125    0.000000         0.0          0.0     0.000657   \n",
       "2        0.001355    0.030497         0.0          0.0     0.000387   \n",
       "3        0.006937    0.000000         0.0          0.0     0.002650   \n",
       "4        0.002290    0.000000         0.0          0.0     0.001359   \n",
       "..            ...         ...         ...          ...          ...   \n",
       "125      0.000000    0.000000         0.0          0.0     0.000000   \n",
       "126      0.000000    0.000000         0.0          0.0     0.060177   \n",
       "127      0.005005    0.016784         0.0          0.0     0.017604   \n",
       "128      0.000000    0.000000         0.0          0.0     0.026844   \n",
       "129      0.000143    0.000000         0.0          0.0     0.002342   \n",
       "\n",
       "     Victivallis  Virgulinella  \n",
       "0            0.0           0.0  \n",
       "1            0.0           0.0  \n",
       "2            0.0           0.0  \n",
       "3            0.0           0.0  \n",
       "4            0.0           0.0  \n",
       "..           ...           ...  \n",
       "125          0.0           0.0  \n",
       "126          0.0           0.0  \n",
       "127          0.0           0.0  \n",
       "128          0.0           0.0  \n",
       "129          0.0           0.0  \n",
       "\n",
       "[130 rows x 168 columns]"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "amplicon_reshaped"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "amplicon_reshaped.to_csv('16S_abundance.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [],
   "source": [
    "chlorobium = amplicon_reshaped[\"Chlorobium\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.0004351203788975447"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "np.mean(chlorobium[chlorobium!= 0])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### looking for the intersection of taxonomy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [],
   "source": [
    "amplicon_taxa_list = set(amplicon_reshaped.columns.values)\n",
    "mgx_taxa_list = set(mgx_reshaped.columns.values)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [],
   "source": [
    "exclude = [\"sampleid\", \"dev_stage\", \"AgeMonths\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [],
   "source": [
    "intersection = [taxa for taxa in amplicon_taxa_list.intersection(mgx_taxa_list) if taxa not in exclude]\n",
    "union = [taxa for taxa in amplicon_taxa_list.union(mgx_taxa_list) if taxa not in exclude]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [],
   "source": [
    "amp_only = [taxa for taxa in amplicon_taxa_list.difference(intersection) if taxa not in exclude]\n",
    "mgx_only = [taxa for taxa in mgx_taxa_list.difference(intersection) if taxa not in exclude]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "105"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/plain": [
       "62"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/plain": [
       "34"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(intersection) # genera found in both\n",
    "len(amp_only) # genera found in amplicon only\n",
    "len(mgx_only) # genera found in mgx only"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1049"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "with open(\"unique_amplicon.txt\", \"w\") as output:\n",
    "    output.write(str(amp_only))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "557"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "with open(\"unique_mgx.txt\", \"w\") as output:\n",
    "    output.write(str(mgx_only))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [],
   "source": [
    "amp = pd.DataFrame({\"taxa\": amp_only})\n",
    "mgx = pd.DataFrame({\"taxa\" : mgx_only})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [],
   "source": [
    "# ITOL formatting, 1 for filled in circle, -1 for no circle\n",
    "\n",
    "amp[\"amp\"] = 1\n",
    "amp[\"mgx\"] = -1\n",
    "mgx[\"amp\"] = -1\n",
    "mgx[\"mgx\"] = 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [],
   "source": [
    "intersection_df = pd.DataFrame({\"taxa\": intersection})\n",
    "intersection_df[\"amp\"] = 1\n",
    "intersection_df[\"mgx\"] = 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [],
   "source": [
    "phylo_mapping = pd.concat([amp, mgx, intersection_df], axis=0, sort=False) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [],
   "source": [
    "database_info = pd.read_csv(\"unique_taxa.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>taxa</th>\n",
       "      <th>in_opp_db</th>\n",
       "      <th>method</th>\n",
       "      <th>few_seqs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Erysipelotrichaceae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>amp</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>Sphingobacterium</td>\n",
       "      <td>NaN</td>\n",
       "      <td>amp</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>Mogibacterium</td>\n",
       "      <td>NaN</td>\n",
       "      <td>amp</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>Lysinibacillus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>amp</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>Planktotalea</td>\n",
       "      <td>NaN</td>\n",
       "      <td>amp</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  taxa  in_opp_db method  few_seqs\n",
       "0  Erysipelotrichaceae        NaN    amp       NaN\n",
       "1     Sphingobacterium        NaN    amp       NaN\n",
       "2        Mogibacterium        NaN    amp       NaN\n",
       "3       Lysinibacillus        NaN    amp       NaN\n",
       "4         Planktotalea        NaN    amp       NaN"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "database_info.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [],
   "source": [
    "# make database dictionary\n",
    "db_dict = {str(s): {} for s in database_info[\"taxa\"]}\n",
    "for index, row in database_info.iterrows():\n",
    "    df_info = row[\"in_opp_db\"]\n",
    "    db_dict[row[\"taxa\"]]= df_info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [],
   "source": [
    "phylo_mapping[\"in_opp_db\"] = phylo_mapping[\"taxa\"].map(db_dict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [],
   "source": [
    "phylo_mapping.fillna(-1, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [],
   "source": [
    "phylo_mapping[\"in_opp_db\"] = phylo_mapping[\"in_opp_db\"].astype(int)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>taxa</th>\n",
       "      <th>amp</th>\n",
       "      <th>mgx</th>\n",
       "      <th>in_opp_db</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Epulopiscium</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>Planktotalea</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>Pseudomonas</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>Oscillospira</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>Bacillus</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>100</td>\n",
       "      <td>Bifidobacterium</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>101</td>\n",
       "      <td>Fusobacterium</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>102</td>\n",
       "      <td>Dorea</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>103</td>\n",
       "      <td>Morganella</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>104</td>\n",
       "      <td>Megamonas</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>201 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                taxa  amp  mgx  in_opp_db\n",
       "0       Epulopiscium    1   -1          1\n",
       "1       Planktotalea    1   -1         -1\n",
       "2        Pseudomonas    1   -1         -1\n",
       "3       Oscillospira    1   -1          1\n",
       "4           Bacillus    1   -1         -1\n",
       "..               ...  ...  ...        ...\n",
       "100  Bifidobacterium    1    1         -1\n",
       "101    Fusobacterium    1    1         -1\n",
       "102            Dorea    1    1         -1\n",
       "103       Morganella    1    1         -1\n",
       "104        Megamonas    1    1         -1\n",
       "\n",
       "[201 rows x 4 columns]"
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "phylo_mapping"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "taxa         object\n",
       "amp           int64\n",
       "mgx           int64\n",
       "in_opp_db     int64\n",
       "dtype: object"
      ]
     },
     "execution_count": 85,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "phylo_mapping.dtypes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [],
   "source": [
    "pd.DataFrame(phylo_mapping).to_csv('/Users/danielle/Documents/thesis/paper-phylogeny/phylo_tree_labels.csv', \n",
    "                                   index=False, header=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [],
   "source": [
    "mgx_list = []\n",
    "amp_list = []"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [],
   "source": [
    "for taxa in union:\n",
    "    if taxa in intersection:\n",
    "        mgx_list.append(\"label1\")\n",
    "        amp_list.append(\"label2\")\n",
    "    else:\n",
    "        if taxa in amp_only:\n",
    "            mgx_list.append(\"lablel1\")\n",
    "            amp_list.append(\"label2\")\n",
    "        elif taxa in mgx_only:\n",
    "            mgx_list.append(\"label1\")\n",
    "            amp_list.append(\"label2\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [],
   "source": [
    "pd.DataFrame(union).to_csv('/Users/danielle/Documents/thesis/paper-phylogeny/all_taxa.txt', index=False, header=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [],
   "source": [
    "pd.DataFrame(union).to_csv('all_taxa.txt', index=False, header=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'union_df' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-74-f157d3aaf0f5>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0munion_df\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"mgx\"\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmgx_list\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      2\u001b[0m \u001b[0munion_df\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"amp\"\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mamp_list\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'union_df' is not defined"
     ]
    }
   ],
   "source": [
    "union_df[\"mgx\"] = mgx_list\n",
    "union_df[\"amp\"] = amp_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "union_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "union_df.to_csv('/Users/danielle/Documents/thesis/paper-phylogeny/all_taxa_methods.txt', \n",
    "             sep='\\t', index = False, header=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### combine two dataframes, long form\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "mgx = mgx.fillna(0)  # fill in missing values with 0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_taxa = pd.merge(amplicon_melt, mgx, on = [\"sampleid\", \"taxa\"], how = \"outer\").fillna(0)\n",
    "merged_taxa[\"mgx_abund\"] = merged_taxa[\"abundance\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_taxa[\"abs_diff\"] = abs(merged_taxa[\"amplicon_abund\"] - merged_taxa[\"mgx_abund\"])\n",
    "merged_taxa[\"tot_diff\"] = (merged_taxa[\"amplicon_abund\"] - merged_taxa[\"mgx_abund\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_taxa[\"AgeMonths\"]= merged_taxa[\"sampleid\"].map(agedict)\n",
    "merged_taxa['dev_stage'] = merged_taxa['AgeMonths'].apply(assign_dev_stage)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# re-order columns\n",
    "\n",
    "cols_to_order = ['sampleid','AgeMonths', 'dev_stage']\n",
    "new_columns = cols_to_order + (merged_taxa.columns.drop(cols_to_order).tolist())\n",
    "merged_taxa = merged_taxa[new_columns]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_taxa.to_csv('paper_abund_df.csv', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## making giant dataframe of abundances"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "amplicon_reshaped[\"method\"] = \"amp\"\n",
    "mgx_reshaped[\"method\"] = \"mgx\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "amplicon_reshaped[\"AgeMonths\"]= amplicon_reshaped[\"sampleid\"].map(agedict)\n",
    "amplicon_reshaped['dev_stage'] = amplicon_reshaped['AgeMonths'].apply(assign_dev_stage)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "amplicon_reshaped"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "concat_df = pd.concat([mgx_reshaped, amplicon_reshaped], sort=True).reset_index(drop = True)\n",
    "concat_df = concat_df.fillna(0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# re-order columns\n",
    "\n",
    "cols_to_order = ['sampleid','AgeMonths', 'dev_stage', \"method\"]\n",
    "new_columns = cols_to_order + (concat_df.columns.drop(cols_to_order).tolist())\n",
    "concat_df = concat_df[new_columns]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "concat_df.sample(15)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "concat_df.to_csv('transposed_mgxamp_df.csv',index=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
