{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/danielle/miniconda2/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: * The project library is out of sync with the lockfile.\n",
      "* Use `renv::restore()` to install packages recorded in the lockfile.\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/Users/danielle/miniconda2/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: Warning:\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/Users/danielle/miniconda2/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning:  namespace ‘ggplot2’ is not available and has been replaced\n",
      "by .GlobalEnv when processing object ‘plot0’\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/Users/danielle/miniconda2/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning:  namespace ‘scales’ is not available and has been replaced\n",
      "by .GlobalEnv when processing object ‘p4’\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n"
     ]
    }
   ],
   "source": [
    "from IPython.core.interactiveshell import InteractiveShell\n",
    "InteractiveShell.ast_node_interactivity = \"all\"\n",
    "\n",
    "import pandas as pd\n",
    "import csv\n",
    "import janitor\n",
    "import numpy as np\n",
    "import rpy2.robjects as robjects"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "metadata = pd.read_csv(\"diabimmune_metadata.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "metadata = metadata.transpose()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "metadata.columns = metadata.iloc[0]\n",
    "metadata.drop([\"Unnamed: 0\"], inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>subjectID</th>\n",
       "      <th>SampleID</th>\n",
       "      <th>age_at_collection</th>\n",
       "      <th>collection_month</th>\n",
       "      <th>delivery</th>\n",
       "      <th>gest_time</th>\n",
       "      <th>gender</th>\n",
       "      <th>country</th>\n",
       "      <th>Exclusive_breast_feeding</th>\n",
       "      <th>Breast_feeding_end</th>\n",
       "      <th>...</th>\n",
       "      <th>allergy_dog</th>\n",
       "      <th>allergy_birch</th>\n",
       "      <th>allergy_timothy</th>\n",
       "      <th>gid_wgs</th>\n",
       "      <th>mgx_reads</th>\n",
       "      <th>mgx_pool</th>\n",
       "      <th>mgx_reads_filtered</th>\n",
       "      <th>read_count_16S</th>\n",
       "      <th>sequencing_PDO_16S</th>\n",
       "      <th>gid_16s</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>E002338</td>\n",
       "      <td>3101193</td>\n",
       "      <td>304</td>\n",
       "      <td>10</td>\n",
       "      <td>vaginal</td>\n",
       "      <td>285</td>\n",
       "      <td>Female</td>\n",
       "      <td>FIN</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>...</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>G80541</td>\n",
       "      <td>28876314</td>\n",
       "      <td>Plate 7</td>\n",
       "      <td>28.568962</td>\n",
       "      <td>15870</td>\n",
       "      <td>PDO-4356</td>\n",
       "      <td>G75694</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>E002338</td>\n",
       "      <td>3107294</td>\n",
       "      <td>493</td>\n",
       "      <td>16</td>\n",
       "      <td>vaginal</td>\n",
       "      <td>285</td>\n",
       "      <td>Female</td>\n",
       "      <td>FIN</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>...</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>G80537</td>\n",
       "      <td>29615868</td>\n",
       "      <td>Plate 7</td>\n",
       "      <td>29.542272</td>\n",
       "      <td>13609</td>\n",
       "      <td>PDO-4356</td>\n",
       "      <td>G75855</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>E002338</td>\n",
       "      <td>3113022</td>\n",
       "      <td>852</td>\n",
       "      <td>28</td>\n",
       "      <td>vaginal</td>\n",
       "      <td>285</td>\n",
       "      <td>Female</td>\n",
       "      <td>FIN</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>...</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>G80322</td>\n",
       "      <td>30478184</td>\n",
       "      <td>Plate 8</td>\n",
       "      <td>30.41078</td>\n",
       "      <td>34620</td>\n",
       "      <td>PDO-4157</td>\n",
       "      <td>G73882</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>E002338</td>\n",
       "      <td>3107293</td>\n",
       "      <td>399</td>\n",
       "      <td>13</td>\n",
       "      <td>vaginal</td>\n",
       "      <td>285</td>\n",
       "      <td>Female</td>\n",
       "      <td>FIN</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>...</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>G80513</td>\n",
       "      <td>28503488</td>\n",
       "      <td>Plate 7</td>\n",
       "      <td>28.360762</td>\n",
       "      <td>7337</td>\n",
       "      <td>PDO-4356</td>\n",
       "      <td>G75849</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>E002338</td>\n",
       "      <td>3101190</td>\n",
       "      <td>212</td>\n",
       "      <td>7</td>\n",
       "      <td>vaginal</td>\n",
       "      <td>285</td>\n",
       "      <td>Female</td>\n",
       "      <td>FIN</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>...</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NaN</td>\n",
       "      <td>23928698</td>\n",
       "      <td>Diab Plate 9</td>\n",
       "      <td>23.824676</td>\n",
       "      <td>18931</td>\n",
       "      <td>PDO-4356</td>\n",
       "      <td>G75788</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 59 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Unnamed: 0 subjectID SampleID age_at_collection collection_month delivery  \\\n",
       "5            E002338  3101193               304               10  vaginal   \n",
       "6            E002338  3107294               493               16  vaginal   \n",
       "7            E002338  3113022               852               28  vaginal   \n",
       "8            E002338  3107293               399               13  vaginal   \n",
       "15           E002338  3101190               212                7  vaginal   \n",
       "\n",
       "Unnamed: 0 gest_time  gender country Exclusive_breast_feeding  \\\n",
       "5                285  Female     FIN                    FALSE   \n",
       "6                285  Female     FIN                    FALSE   \n",
       "7                285  Female     FIN                    FALSE   \n",
       "8                285  Female     FIN                    FALSE   \n",
       "15               285  Female     FIN                    FALSE   \n",
       "\n",
       "Unnamed: 0 Breast_feeding_end  ... allergy_dog allergy_birch allergy_timothy  \\\n",
       "5                       FALSE  ...       FALSE         FALSE           FALSE   \n",
       "6                        TRUE  ...       FALSE         FALSE           FALSE   \n",
       "7                        TRUE  ...       FALSE         FALSE           FALSE   \n",
       "8                       FALSE  ...       FALSE         FALSE           FALSE   \n",
       "15                      FALSE  ...       FALSE         FALSE           FALSE   \n",
       "\n",
       "Unnamed: 0 gid_wgs mgx_reads      mgx_pool mgx_reads_filtered read_count_16S  \\\n",
       "5           G80541  28876314       Plate 7          28.568962          15870   \n",
       "6           G80537  29615868       Plate 7          29.542272          13609   \n",
       "7           G80322  30478184       Plate 8           30.41078          34620   \n",
       "8           G80513  28503488       Plate 7          28.360762           7337   \n",
       "15             NaN  23928698  Diab Plate 9          23.824676          18931   \n",
       "\n",
       "Unnamed: 0 sequencing_PDO_16S gid_16s  \n",
       "5                    PDO-4356  G75694  \n",
       "6                    PDO-4356  G75855  \n",
       "7                    PDO-4157  G73882  \n",
       "8                    PDO-4356  G75849  \n",
       "15                   PDO-4356  G75788  \n",
       "\n",
       "[5 rows x 59 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metadata.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "# map gid_16s to gid_wgs\n",
    "# keys are 16S ids, values are mgx ids\n",
    "\n",
    "id_mapping = pd.Series(metadata.gid_wgs.values,index=metadata.gid_16s).to_dict()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Importing metagenomics data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Genus "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# read in mgx data\n",
    "mgx_genus = pd.read_csv(\"diabimmune_mgx_genus.csv\")\n",
    "mgx_genus.rename(columns={'taxname': 'taxa'}, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "# mgx_genus = mgx[mgx['taxa'].str.contains(\"\\|g__\")] # keep genera\n",
    "# mgx_genus = mgx_genus[~mgx_genus['taxa'].str.contains(\"\\|s__\")] # keep species\n",
    "# mgx_genus[\"taxa\"] = mgx_genus['taxa'].str.split(\"\\|g__\").str[-1]\n",
    "# mgx_genus[\"taxa\"] = mgx_genus['taxa'].str.split(\"\\|s__\").str[0]#"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "# remove taxa that are unclassified or have no name\n",
    "# \"_unclassified\"\n",
    "# \"_noname\"\n",
    "mgx_genus = mgx_genus[~mgx_genus.taxa.str.contains(\"_unclassified\")]\n",
    "mgx_genus = mgx_genus[~mgx_genus.taxa.str.contains(\"_noname\")]\n",
    "mgx_genus = mgx_genus[~mgx_genus.taxa.str.contains(\"virus\")]\n",
    "mgx_genus = mgx_genus[~mgx_genus.taxa.str.contains(\"Candidatus\")]\n",
    "mgx_genus = mgx_genus[~mgx_genus.taxa.str.contains(\"candidate\")]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "mgx_genus = mgx_genus.groupby(['taxa']).sum().reset_index() "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>taxa</th>\n",
       "      <th>G69146</th>\n",
       "      <th>G69147</th>\n",
       "      <th>G69148</th>\n",
       "      <th>G69149</th>\n",
       "      <th>G69150</th>\n",
       "      <th>G69151</th>\n",
       "      <th>G69152</th>\n",
       "      <th>G69153</th>\n",
       "      <th>G69154</th>\n",
       "      <th>...</th>\n",
       "      <th>G80617</th>\n",
       "      <th>G80619</th>\n",
       "      <th>G80620</th>\n",
       "      <th>G80621</th>\n",
       "      <th>G80623</th>\n",
       "      <th>G80624</th>\n",
       "      <th>G86980</th>\n",
       "      <th>G86986</th>\n",
       "      <th>G90143</th>\n",
       "      <th>G90147</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Absiella</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Acidaminococcus</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.013550</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Acinetobacter</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Actinobaculum</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001091</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Actinomyces</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000045</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000083</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000766</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000011</td>\n",
       "      <td>0.000051</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 801 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              taxa  G69146  G69147  G69148  G69149  G69150  G69151    G69152  \\\n",
       "0         Absiella     0.0     0.0     0.0     0.0     0.0     0.0  0.000000   \n",
       "1  Acidaminococcus     0.0     0.0     0.0     0.0     0.0     0.0  0.000000   \n",
       "2    Acinetobacter     0.0     0.0     0.0     0.0     0.0     0.0  0.000000   \n",
       "3    Actinobaculum     0.0     0.0     0.0     0.0     0.0     0.0  0.000000   \n",
       "4      Actinomyces     0.0     0.0     0.0     0.0     0.0     0.0  0.000045   \n",
       "\n",
       "   G69153    G69154  ...  G80617  G80619  G80620  G80621  G80623    G80624  \\\n",
       "0     0.0  0.000000  ...     0.0     0.0     0.0     0.0     0.0  0.000000   \n",
       "1     0.0  0.013550  ...     0.0     0.0     0.0     0.0     0.0  0.000000   \n",
       "2     0.0  0.000000  ...     0.0     0.0     0.0     0.0     0.0  0.000000   \n",
       "3     0.0  0.000000  ...     0.0     0.0     0.0     0.0     0.0  0.000000   \n",
       "4     0.0  0.000083  ...     0.0     0.0     0.0     0.0     0.0  0.000766   \n",
       "\n",
       "   G86980  G86986    G90143    G90147  \n",
       "0     0.0     0.0  0.000000  0.000000  \n",
       "1     0.0     0.0  0.000000  0.000000  \n",
       "2     0.0     0.0  0.000000  0.000000  \n",
       "3     0.0     0.0  0.000000  0.001091  \n",
       "4     0.0     0.0  0.000011  0.000051  \n",
       "\n",
       "[5 rows x 801 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mgx_genus.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Families"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "mgx_family = pd.read_csv(\"diabimmune_mgx_family.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>taxname</th>\n",
       "      <th>G69146</th>\n",
       "      <th>G69147</th>\n",
       "      <th>G69148</th>\n",
       "      <th>G69149</th>\n",
       "      <th>G69150</th>\n",
       "      <th>G69151</th>\n",
       "      <th>G69152</th>\n",
       "      <th>G69153</th>\n",
       "      <th>G69154</th>\n",
       "      <th>...</th>\n",
       "      <th>G80617</th>\n",
       "      <th>G80619</th>\n",
       "      <th>G80620</th>\n",
       "      <th>G80621</th>\n",
       "      <th>G80623</th>\n",
       "      <th>G80624</th>\n",
       "      <th>G86980</th>\n",
       "      <th>G86986</th>\n",
       "      <th>G90143</th>\n",
       "      <th>G90147</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Enterobacteriaceae</td>\n",
       "      <td>0.466080</td>\n",
       "      <td>0.327810</td>\n",
       "      <td>0.012988</td>\n",
       "      <td>0.045543</td>\n",
       "      <td>0.020078</td>\n",
       "      <td>0.042352</td>\n",
       "      <td>0.055992</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.003681</td>\n",
       "      <td>...</td>\n",
       "      <td>0.196305</td>\n",
       "      <td>0.021126</td>\n",
       "      <td>0.037293</td>\n",
       "      <td>0.751990</td>\n",
       "      <td>0.000040</td>\n",
       "      <td>0.001803</td>\n",
       "      <td>0.498086</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.006707</td>\n",
       "      <td>0.023033</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Akkermansiaceae</td>\n",
       "      <td>0.242453</td>\n",
       "      <td>0.005630</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.079431</td>\n",
       "      <td>0.052597</td>\n",
       "      <td>0.003702</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000022</td>\n",
       "      <td>0.883796</td>\n",
       "      <td>0.186946</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002018</td>\n",
       "      <td>0.573473</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Bacteroidaceae</td>\n",
       "      <td>0.066054</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.097642</td>\n",
       "      <td>0.228180</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.496944</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.586511</td>\n",
       "      <td>0.498754</td>\n",
       "      <td>...</td>\n",
       "      <td>0.615055</td>\n",
       "      <td>0.000142</td>\n",
       "      <td>0.001016</td>\n",
       "      <td>0.017285</td>\n",
       "      <td>0.538860</td>\n",
       "      <td>0.054789</td>\n",
       "      <td>0.075458</td>\n",
       "      <td>0.138963</td>\n",
       "      <td>0.829793</td>\n",
       "      <td>0.000027</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Lachnospiraceae</td>\n",
       "      <td>0.060958</td>\n",
       "      <td>0.343531</td>\n",
       "      <td>0.457044</td>\n",
       "      <td>0.130974</td>\n",
       "      <td>0.698545</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.049892</td>\n",
       "      <td>0.081940</td>\n",
       "      <td>0.009887</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.062178</td>\n",
       "      <td>0.822903</td>\n",
       "      <td>0.001328</td>\n",
       "      <td>0.152402</td>\n",
       "      <td>0.011259</td>\n",
       "      <td>0.074158</td>\n",
       "      <td>0.031986</td>\n",
       "      <td>0.093438</td>\n",
       "      <td>0.000593</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Veillonellaceae</td>\n",
       "      <td>0.058309</td>\n",
       "      <td>0.008770</td>\n",
       "      <td>0.015439</td>\n",
       "      <td>0.022370</td>\n",
       "      <td>0.002777</td>\n",
       "      <td>0.230625</td>\n",
       "      <td>0.195566</td>\n",
       "      <td>0.000180</td>\n",
       "      <td>0.019004</td>\n",
       "      <td>...</td>\n",
       "      <td>0.120381</td>\n",
       "      <td>0.089666</td>\n",
       "      <td>0.060263</td>\n",
       "      <td>0.151497</td>\n",
       "      <td>0.002519</td>\n",
       "      <td>0.000010</td>\n",
       "      <td>0.032850</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.005980</td>\n",
       "      <td>0.342829</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 801 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              taxname    G69146    G69147    G69148    G69149    G69150  \\\n",
       "0  Enterobacteriaceae  0.466080  0.327810  0.012988  0.045543  0.020078   \n",
       "1     Akkermansiaceae  0.242453  0.005630  0.000000  0.000000  0.000000   \n",
       "2      Bacteroidaceae  0.066054  0.000000  0.097642  0.228180  0.000000   \n",
       "3     Lachnospiraceae  0.060958  0.343531  0.457044  0.130974  0.698545   \n",
       "4     Veillonellaceae  0.058309  0.008770  0.015439  0.022370  0.002777   \n",
       "\n",
       "     G69151    G69152    G69153    G69154  ...    G80617    G80619    G80620  \\\n",
       "0  0.042352  0.055992  0.000000  0.003681  ...  0.196305  0.021126  0.037293   \n",
       "1  0.000000  0.079431  0.052597  0.003702  ...  0.000000  0.000000  0.000000   \n",
       "2  0.496944  0.000000  0.586511  0.498754  ...  0.615055  0.000142  0.001016   \n",
       "3  0.000000  0.049892  0.081940  0.009887  ...  0.000000  0.062178  0.822903   \n",
       "4  0.230625  0.195566  0.000180  0.019004  ...  0.120381  0.089666  0.060263   \n",
       "\n",
       "     G80621    G80623    G80624    G86980    G86986    G90143    G90147  \n",
       "0  0.751990  0.000040  0.001803  0.498086  0.000000  0.006707  0.023033  \n",
       "1  0.000000  0.000022  0.883796  0.186946  0.000000  0.002018  0.573473  \n",
       "2  0.017285  0.538860  0.054789  0.075458  0.138963  0.829793  0.000027  \n",
       "3  0.001328  0.152402  0.011259  0.074158  0.031986  0.093438  0.000593  \n",
       "4  0.151497  0.002519  0.000010  0.032850  0.000000  0.005980  0.342829  \n",
       "\n",
       "[5 rows x 801 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mgx_family.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Looking at species in metagenomic data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "# mgx_species = mgx[mgx['taxa'].str.contains(\"\\|s__\")] # keep families genera\n",
    "# mgx_species = mgx_species[~mgx_species['taxa'].str.contains(\"\\|t__\")] # remove genera\n",
    "\n",
    "# mgx_species[\"taxa\"] = mgx_species['taxa'].str.split(\"\\|s__\").str[-1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "# mgx_species = mgx_species[~mgx_species.taxa.str.contains(\"_unclassified\")]\n",
    "# mgx_species = mgx_species[~mgx_species.taxa.str.contains(\"_noname\")]\n",
    "# mgx_species = mgx_species[~mgx_species.taxa.str.contains(\"virus\")]\n",
    "# mgx_species = mgx_species[~mgx_species.taxa.str.contains(\"Candidatus\")]\n",
    "# mgx_species = mgx_species[~mgx_species.taxa.str.contains(\"candidate\")]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "# mgx_species = mgx_species.groupby(['taxa']).sum().reset_index() "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "mgx_species = pd.read_csv(\"diabimmune_mgx_species.csv\")\n",
    "# mgx_genus.rename(columns={'taxname': 'taxa'}, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>taxname</th>\n",
       "      <th>G69146</th>\n",
       "      <th>G69147</th>\n",
       "      <th>G69148</th>\n",
       "      <th>G69149</th>\n",
       "      <th>G69150</th>\n",
       "      <th>G69151</th>\n",
       "      <th>G69152</th>\n",
       "      <th>G69153</th>\n",
       "      <th>G69154</th>\n",
       "      <th>...</th>\n",
       "      <th>G80617</th>\n",
       "      <th>G80619</th>\n",
       "      <th>G80620</th>\n",
       "      <th>G80621</th>\n",
       "      <th>G80623</th>\n",
       "      <th>G80624</th>\n",
       "      <th>G86980</th>\n",
       "      <th>G86986</th>\n",
       "      <th>G90143</th>\n",
       "      <th>G90147</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Escherichia_coli</td>\n",
       "      <td>0.284918</td>\n",
       "      <td>0.32781</td>\n",
       "      <td>0.004385</td>\n",
       "      <td>0.045543</td>\n",
       "      <td>0.020078</td>\n",
       "      <td>0.042352</td>\n",
       "      <td>0.050934</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.003681</td>\n",
       "      <td>...</td>\n",
       "      <td>0.196305</td>\n",
       "      <td>0.000056</td>\n",
       "      <td>0.036164</td>\n",
       "      <td>0.617818</td>\n",
       "      <td>0.000040</td>\n",
       "      <td>0.000136</td>\n",
       "      <td>0.319663</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.006707</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Akkermansia_muciniphila</td>\n",
       "      <td>0.242453</td>\n",
       "      <td>0.00563</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.079431</td>\n",
       "      <td>0.052597</td>\n",
       "      <td>0.003702</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000022</td>\n",
       "      <td>0.883796</td>\n",
       "      <td>0.186946</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002018</td>\n",
       "      <td>0.573473</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Klebsiella_pneumoniae</td>\n",
       "      <td>0.133182</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001958</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.132023</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.015799</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Dialister_invisus</td>\n",
       "      <td>0.057785</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.015439</td>\n",
       "      <td>0.000078</td>\n",
       "      <td>0.002291</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000010</td>\n",
       "      <td>0.032804</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001019</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Bifidobacterium_longum</td>\n",
       "      <td>0.041493</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.012710</td>\n",
       "      <td>0.119270</td>\n",
       "      <td>0.005591</td>\n",
       "      <td>0.171838</td>\n",
       "      <td>0.186589</td>\n",
       "      <td>0.009908</td>\n",
       "      <td>0.050695</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.011101</td>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.000713</td>\n",
       "      <td>0.167309</td>\n",
       "      <td>0.006347</td>\n",
       "      <td>0.056740</td>\n",
       "      <td>0.344709</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.021620</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 801 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   taxname    G69146   G69147    G69148    G69149    G69150  \\\n",
       "0         Escherichia_coli  0.284918  0.32781  0.004385  0.045543  0.020078   \n",
       "1  Akkermansia_muciniphila  0.242453  0.00563  0.000000  0.000000  0.000000   \n",
       "2    Klebsiella_pneumoniae  0.133182  0.00000  0.000000  0.000000  0.000000   \n",
       "3        Dialister_invisus  0.057785  0.00000  0.015439  0.000078  0.002291   \n",
       "4   Bifidobacterium_longum  0.041493  0.00000  0.012710  0.119270  0.005591   \n",
       "\n",
       "     G69151    G69152    G69153    G69154  ...    G80617    G80619    G80620  \\\n",
       "0  0.042352  0.050934  0.000000  0.003681  ...  0.196305  0.000056  0.036164   \n",
       "1  0.000000  0.079431  0.052597  0.003702  ...  0.000000  0.000000  0.000000   \n",
       "2  0.000000  0.000000  0.000000  0.000000  ...  0.000000  0.000000  0.000000   \n",
       "3  0.000000  0.000000  0.000000  0.000000  ...  0.000000  0.000000  0.000000   \n",
       "4  0.171838  0.186589  0.009908  0.050695  ...  0.000000  0.011101  0.000200   \n",
       "\n",
       "     G80621    G80623    G80624    G86980    G86986    G90143    G90147  \n",
       "0  0.617818  0.000040  0.000136  0.319663  0.000000  0.006707  0.000000  \n",
       "1  0.000000  0.000022  0.883796  0.186946  0.000000  0.002018  0.573473  \n",
       "2  0.001958  0.000000  0.000000  0.132023  0.000000  0.000000  0.015799  \n",
       "3  0.000000  0.000000  0.000010  0.032804  0.000000  0.001019  0.000000  \n",
       "4  0.000713  0.167309  0.006347  0.056740  0.344709  0.000000  0.021620  \n",
       "\n",
       "[5 rows x 801 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mgx_species.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "511"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(mgx_species)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Importing amplicon data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Genus"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "# read in 16S data\n",
    "\n",
    "amp_genus = pd.read_csv(\"karalia_dada2_genera.csv\")\n",
    "amp_genus.rename(columns={'genus': 'taxa'}, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>taxa</th>\n",
       "      <th>G63225</th>\n",
       "      <th>G63227</th>\n",
       "      <th>G63228</th>\n",
       "      <th>G63229</th>\n",
       "      <th>G63230</th>\n",
       "      <th>G63231</th>\n",
       "      <th>G63232</th>\n",
       "      <th>G63234</th>\n",
       "      <th>G63236</th>\n",
       "      <th>...</th>\n",
       "      <th>G76485</th>\n",
       "      <th>G76486</th>\n",
       "      <th>G76487</th>\n",
       "      <th>G76488</th>\n",
       "      <th>G76489</th>\n",
       "      <th>G76490</th>\n",
       "      <th>G76491</th>\n",
       "      <th>G76492</th>\n",
       "      <th>G76493</th>\n",
       "      <th>G76494</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Bifidobacterium</td>\n",
       "      <td>0.004997</td>\n",
       "      <td>0.010805</td>\n",
       "      <td>0.007308</td>\n",
       "      <td>0.000231</td>\n",
       "      <td>0.003715</td>\n",
       "      <td>0.007665</td>\n",
       "      <td>0.104364</td>\n",
       "      <td>0.027547</td>\n",
       "      <td>0.014981</td>\n",
       "      <td>...</td>\n",
       "      <td>0.230535</td>\n",
       "      <td>0.041893</td>\n",
       "      <td>0.248718</td>\n",
       "      <td>0.003288</td>\n",
       "      <td>0.235009</td>\n",
       "      <td>0.237299</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.860111</td>\n",
       "      <td>0.687500</td>\n",
       "      <td>0.012104</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Bacteroides</td>\n",
       "      <td>0.436124</td>\n",
       "      <td>0.343189</td>\n",
       "      <td>0.070793</td>\n",
       "      <td>0.059532</td>\n",
       "      <td>0.304362</td>\n",
       "      <td>0.183758</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.019480</td>\n",
       "      <td>0.150320</td>\n",
       "      <td>...</td>\n",
       "      <td>0.308592</td>\n",
       "      <td>0.151767</td>\n",
       "      <td>0.360050</td>\n",
       "      <td>0.040434</td>\n",
       "      <td>0.324846</td>\n",
       "      <td>0.480689</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000242</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>UNCLASSIFIED</td>\n",
       "      <td>0.146095</td>\n",
       "      <td>0.031704</td>\n",
       "      <td>0.195331</td>\n",
       "      <td>0.224378</td>\n",
       "      <td>0.147593</td>\n",
       "      <td>0.510983</td>\n",
       "      <td>0.054241</td>\n",
       "      <td>0.201836</td>\n",
       "      <td>0.310966</td>\n",
       "      <td>...</td>\n",
       "      <td>0.322257</td>\n",
       "      <td>0.640546</td>\n",
       "      <td>0.136235</td>\n",
       "      <td>0.693582</td>\n",
       "      <td>0.139440</td>\n",
       "      <td>0.240539</td>\n",
       "      <td>0.473730</td>\n",
       "      <td>0.048752</td>\n",
       "      <td>0.031179</td>\n",
       "      <td>0.764585</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Faecalibacterium</td>\n",
       "      <td>0.036886</td>\n",
       "      <td>0.016238</td>\n",
       "      <td>0.113969</td>\n",
       "      <td>0.035317</td>\n",
       "      <td>0.230022</td>\n",
       "      <td>0.039644</td>\n",
       "      <td>0.000433</td>\n",
       "      <td>0.227238</td>\n",
       "      <td>0.007121</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.025008</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Prevotella</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.070750</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000092</td>\n",
       "      <td>0.001560</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.024074</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000688</td>\n",
       "      <td>0.003179</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.005573</td>\n",
       "      <td>0.000861</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 1836 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               taxa    G63225    G63227    G63228    G63229    G63230  \\\n",
       "0   Bifidobacterium  0.004997  0.010805  0.007308  0.000231  0.003715   \n",
       "1       Bacteroides  0.436124  0.343189  0.070793  0.059532  0.304362   \n",
       "2      UNCLASSIFIED  0.146095  0.031704  0.195331  0.224378  0.147593   \n",
       "3  Faecalibacterium  0.036886  0.016238  0.113969  0.035317  0.230022   \n",
       "4        Prevotella  0.000000  0.000000  0.070750  0.000000  0.000000   \n",
       "\n",
       "     G63231    G63232    G63234    G63236  ...    G76485    G76486    G76487  \\\n",
       "0  0.007665  0.104364  0.027547  0.014981  ...  0.230535  0.041893  0.248718   \n",
       "1  0.183758  0.000000  0.019480  0.150320  ...  0.308592  0.151767  0.360050   \n",
       "2  0.510983  0.054241  0.201836  0.310966  ...  0.322257  0.640546  0.136235   \n",
       "3  0.039644  0.000433  0.227238  0.007121  ...  0.000000  0.000000  0.025008   \n",
       "4  0.000092  0.001560  0.000000  0.024074  ...  0.000688  0.003179  0.000000   \n",
       "\n",
       "     G76488    G76489    G76490    G76491    G76492    G76493    G76494  \n",
       "0  0.003288  0.235009  0.237299  0.000000  0.860111  0.687500  0.012104  \n",
       "1  0.040434  0.324846  0.480689  0.000000  0.000000  0.000000  0.000242  \n",
       "2  0.693582  0.139440  0.240539  0.473730  0.048752  0.031179  0.764585  \n",
       "3  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  \n",
       "4  0.000000  0.000000  0.005573  0.000861  0.000000  0.000000  0.000000  \n",
       "\n",
       "[5 rows x 1836 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "amp_genus.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "# remove taxa that are unclassified or have no name\n",
    "# \"_unclassified\"\n",
    "# \"_noname\"\n",
    "amp_genus = amp_genus[~amp_genus.taxa.str.contains(\"_unclassified\")]\n",
    "amp_genus = amp_genus[~amp_genus.taxa.str.contains(\"_noname\")]\n",
    "amp_genus = amp_genus[~amp_genus.taxa.str.contains(\"virus\")]\n",
    "amp_genus = amp_genus[~amp_genus.taxa.str.contains(\"Candidatus\")]\n",
    "amp_genus = amp_genus[~amp_genus.taxa.str.contains(r'[0-9]')]\n",
    "amp_genus = amp_genus[~amp_genus.taxa.str.contains(\"group\")]\n",
    "amp_genus['taxa'].replace('', np.nan, inplace=True)\n",
    "amp_genus.dropna(subset=['taxa'], inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "amp_genus.taxa = amp_genus.taxa = amp_genus.taxa.str.strip('[]')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "# remove digits\n",
    "amp_genus.taxa = amp_genus[~amp_genus.taxa.str.contains(r'\\d')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "amp_genus = amp_genus.groupby(['taxa']).sum().reset_index() "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "amp_genus = pd.DataFrame(amp_genus).rename(columns=id_mapping)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "amp_genus = amp_genus.loc[:, amp_genus.columns.notnull()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>taxa</th>\n",
       "      <th>G78487</th>\n",
       "      <th>G78488</th>\n",
       "      <th>G78489</th>\n",
       "      <th>G78490</th>\n",
       "      <th>G78492</th>\n",
       "      <th>G78493</th>\n",
       "      <th>G78494</th>\n",
       "      <th>G78495</th>\n",
       "      <th>G78496</th>\n",
       "      <th>...</th>\n",
       "      <th>G80605</th>\n",
       "      <th>G80594</th>\n",
       "      <th>G80611</th>\n",
       "      <th>G80617</th>\n",
       "      <th>G80623</th>\n",
       "      <th>G80556</th>\n",
       "      <th>G80563</th>\n",
       "      <th>G80570</th>\n",
       "      <th>G80576</th>\n",
       "      <th>G80582</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Abiotrophia</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Acetanaerobacterium</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000187</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Acetobacter</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Achromobacter</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Acidaminococcus</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 842 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  taxa  G78487    G78488  G78489  G78490  G78492  G78493  \\\n",
       "0          Abiotrophia     0.0  0.000000     0.0     0.0     0.0     0.0   \n",
       "1  Acetanaerobacterium     0.0  0.000187     0.0     0.0     0.0     0.0   \n",
       "2          Acetobacter     0.0  0.000000     0.0     0.0     0.0     0.0   \n",
       "3        Achromobacter     0.0  0.000000     0.0     0.0     0.0     0.0   \n",
       "4      Acidaminococcus     0.0  0.000000     0.0     0.0     0.0     0.0   \n",
       "\n",
       "   G78494  G78495  G78496  ...  G80605  G80594  G80611  G80617  G80623  \\\n",
       "0     0.0     0.0     0.0  ...     0.0     0.0     0.0     0.0     0.0   \n",
       "1     0.0     0.0     0.0  ...     0.0     0.0     0.0     0.0     0.0   \n",
       "2     0.0     0.0     0.0  ...     0.0     0.0     0.0     0.0     0.0   \n",
       "3     0.0     0.0     0.0  ...     0.0     0.0     0.0     0.0     0.0   \n",
       "4     0.0     0.0     0.0  ...     0.0     0.0     0.0     0.0     0.0   \n",
       "\n",
       "   G80556  G80563  G80570  G80576  G80582  \n",
       "0     0.0     0.0     0.0     0.0     0.0  \n",
       "1     0.0     0.0     0.0     0.0     0.0  \n",
       "2     0.0     0.0     0.0     0.0     0.0  \n",
       "3     0.0     0.0     0.0     0.0     0.0  \n",
       "4     0.0     0.0     0.0     0.0     0.0  \n",
       "\n",
       "[5 rows x 842 columns]"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "amp_genus.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(302, 842)"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "amp_genus.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(198, 801)"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mgx_genus.shape"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### amplicon family"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "# amp_family = amp[amp['taxa'].str.contains(\"\\|f__\")] # keep families genera\n",
    "# amp_family = amp_family[~amp_family['taxa'].str.contains(\"\\|g__\")] # remove genera\n",
    "\n",
    "# amp_family[\"taxa\"] = amp_family['taxa'].str.split(\"\\|f__\").str[-1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "# amp_family = amp_family[~amp_family.taxa.str.contains(\"_unclassified\")]\n",
    "# amp_family = amp_family[~amp_family.taxa.str.contains(\"_noname\")]\n",
    "# amp_family = amp_family[~amp_family.taxa.str.contains(\"virus\")]\n",
    "# amp_family = amp_family[~amp_family.taxa.str.contains(\"Candidatus\")]\n",
    "# amp_family = amp_family[~amp_family.taxa.str.contains(r'[0-9]')]\n",
    "# amp_family = amp_family[~amp_family.taxa.str.contains(\"group\")]\n",
    "# amp_family['taxa'].replace('', np.nan, inplace=True)\n",
    "# amp_family.dropna(subset=['taxa'], inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "# amp_family = amp_family.groupby(['taxa']).sum().reset_index() "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "# amp_family.sort_values(by = \"taxa\", ascending=True, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "amp_family = pd.read_csv(\"karalia_dada2_family.csv\")\n",
    "amp_family.rename(columns={'family': 'taxa'}, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>taxa</th>\n",
       "      <th>G63225</th>\n",
       "      <th>G63227</th>\n",
       "      <th>G63228</th>\n",
       "      <th>G63229</th>\n",
       "      <th>G63230</th>\n",
       "      <th>G63231</th>\n",
       "      <th>G63232</th>\n",
       "      <th>G63234</th>\n",
       "      <th>G63236</th>\n",
       "      <th>...</th>\n",
       "      <th>G76485</th>\n",
       "      <th>G76486</th>\n",
       "      <th>G76487</th>\n",
       "      <th>G76488</th>\n",
       "      <th>G76489</th>\n",
       "      <th>G76490</th>\n",
       "      <th>G76491</th>\n",
       "      <th>G76492</th>\n",
       "      <th>G76493</th>\n",
       "      <th>G76494</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Bifidobacteriaceae</td>\n",
       "      <td>0.004997</td>\n",
       "      <td>0.010805</td>\n",
       "      <td>0.007308</td>\n",
       "      <td>0.000231</td>\n",
       "      <td>0.003715</td>\n",
       "      <td>0.007665</td>\n",
       "      <td>0.104364</td>\n",
       "      <td>0.027547</td>\n",
       "      <td>0.014981</td>\n",
       "      <td>...</td>\n",
       "      <td>0.230535</td>\n",
       "      <td>0.041893</td>\n",
       "      <td>0.248718</td>\n",
       "      <td>0.003288</td>\n",
       "      <td>0.235009</td>\n",
       "      <td>0.237299</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.860111</td>\n",
       "      <td>0.687500</td>\n",
       "      <td>0.012104</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Bacteroidaceae</td>\n",
       "      <td>0.436124</td>\n",
       "      <td>0.343189</td>\n",
       "      <td>0.070793</td>\n",
       "      <td>0.059532</td>\n",
       "      <td>0.304362</td>\n",
       "      <td>0.183758</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.019480</td>\n",
       "      <td>0.150320</td>\n",
       "      <td>...</td>\n",
       "      <td>0.308592</td>\n",
       "      <td>0.151767</td>\n",
       "      <td>0.360050</td>\n",
       "      <td>0.040434</td>\n",
       "      <td>0.324846</td>\n",
       "      <td>0.480689</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000242</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Enterobacteriaceae</td>\n",
       "      <td>0.001728</td>\n",
       "      <td>0.000386</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.200328</td>\n",
       "      <td>0.000362</td>\n",
       "      <td>0.020244</td>\n",
       "      <td>0.001040</td>\n",
       "      <td>0.000343</td>\n",
       "      <td>0.284879</td>\n",
       "      <td>...</td>\n",
       "      <td>0.320881</td>\n",
       "      <td>0.587339</td>\n",
       "      <td>0.096265</td>\n",
       "      <td>0.590793</td>\n",
       "      <td>0.121032</td>\n",
       "      <td>0.237558</td>\n",
       "      <td>0.431094</td>\n",
       "      <td>0.048752</td>\n",
       "      <td>0.031179</td>\n",
       "      <td>0.736747</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Ruminococcaceae</td>\n",
       "      <td>0.137221</td>\n",
       "      <td>0.043340</td>\n",
       "      <td>0.217802</td>\n",
       "      <td>0.036878</td>\n",
       "      <td>0.311303</td>\n",
       "      <td>0.081439</td>\n",
       "      <td>0.002709</td>\n",
       "      <td>0.293401</td>\n",
       "      <td>0.014379</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002151</td>\n",
       "      <td>0.113843</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000772</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Prevotellaceae</td>\n",
       "      <td>0.000058</td>\n",
       "      <td>0.006946</td>\n",
       "      <td>0.070750</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.014278</td>\n",
       "      <td>0.001560</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.024074</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000688</td>\n",
       "      <td>0.003179</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.005573</td>\n",
       "      <td>0.000861</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 1836 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 taxa    G63225    G63227    G63228    G63229    G63230  \\\n",
       "0  Bifidobacteriaceae  0.004997  0.010805  0.007308  0.000231  0.003715   \n",
       "1      Bacteroidaceae  0.436124  0.343189  0.070793  0.059532  0.304362   \n",
       "2  Enterobacteriaceae  0.001728  0.000386  0.000000  0.200328  0.000362   \n",
       "3     Ruminococcaceae  0.137221  0.043340  0.217802  0.036878  0.311303   \n",
       "4      Prevotellaceae  0.000058  0.006946  0.070750  0.000000  0.000000   \n",
       "\n",
       "     G63231    G63232    G63234    G63236  ...    G76485    G76486    G76487  \\\n",
       "0  0.007665  0.104364  0.027547  0.014981  ...  0.230535  0.041893  0.248718   \n",
       "1  0.183758  0.000000  0.019480  0.150320  ...  0.308592  0.151767  0.360050   \n",
       "2  0.020244  0.001040  0.000343  0.284879  ...  0.320881  0.587339  0.096265   \n",
       "3  0.081439  0.002709  0.293401  0.014379  ...  0.000000  0.002151  0.113843   \n",
       "4  0.014278  0.001560  0.000000  0.024074  ...  0.000688  0.003179  0.000000   \n",
       "\n",
       "     G76488    G76489    G76490    G76491    G76492    G76493    G76494  \n",
       "0  0.003288  0.235009  0.237299  0.000000  0.860111  0.687500  0.012104  \n",
       "1  0.040434  0.324846  0.480689  0.000000  0.000000  0.000000  0.000242  \n",
       "2  0.590793  0.121032  0.237558  0.431094  0.048752  0.031179  0.736747  \n",
       "3  0.000000  0.000772  0.000000  0.000000  0.000000  0.000000  0.000000  \n",
       "4  0.000000  0.000000  0.005573  0.000861  0.000000  0.000000  0.000000  \n",
       "\n",
       "[5 rows x 1836 columns]"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "amp_family.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "# ratio of unclassified genera for 16S profiling\n",
    "# sum(amplicon.taxa.str.contains(\"_unclassified| unidentified| uncultured| group\"))/amplicon.shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "# amplicon.to_csv('16S_abundance.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### amplicon species"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "amp_species = pd.read_csv(\"karalia_dada2_species.csv\")\n",
    "amp_species.rename(columns={'species': 'taxa'}, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "amp_species[\"taxa\"] = amp_species['taxa'].str.replace('s__', '')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>taxa</th>\n",
       "      <th>G63225</th>\n",
       "      <th>G63227</th>\n",
       "      <th>G63228</th>\n",
       "      <th>G63229</th>\n",
       "      <th>G63230</th>\n",
       "      <th>G63231</th>\n",
       "      <th>G63232</th>\n",
       "      <th>G63234</th>\n",
       "      <th>G63236</th>\n",
       "      <th>...</th>\n",
       "      <th>G76485</th>\n",
       "      <th>G76486</th>\n",
       "      <th>G76487</th>\n",
       "      <th>G76488</th>\n",
       "      <th>G76489</th>\n",
       "      <th>G76490</th>\n",
       "      <th>G76491</th>\n",
       "      <th>G76492</th>\n",
       "      <th>G76493</th>\n",
       "      <th>G76494</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>UNCLASSIFIED</td>\n",
       "      <td>0.958397</td>\n",
       "      <td>0.413988</td>\n",
       "      <td>0.716716</td>\n",
       "      <td>0.968321</td>\n",
       "      <td>0.838273</td>\n",
       "      <td>0.950505</td>\n",
       "      <td>0.680969</td>\n",
       "      <td>0.944478</td>\n",
       "      <td>0.696196</td>\n",
       "      <td>...</td>\n",
       "      <td>0.929316</td>\n",
       "      <td>0.910137</td>\n",
       "      <td>0.896830</td>\n",
       "      <td>0.905858</td>\n",
       "      <td>0.740300</td>\n",
       "      <td>0.980560</td>\n",
       "      <td>0.830749</td>\n",
       "      <td>0.993131</td>\n",
       "      <td>0.954571</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Bifidobacterium_bifidum</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001202</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000312</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.049548</td>\n",
       "      <td>0.015803</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.011534</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.006869</td>\n",
       "      <td>0.045429</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Akkermansia_muciniphila</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.519963</td>\n",
       "      <td>0.205563</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.006350</td>\n",
       "      <td>0.272486</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Ruminococcus_gnavus</td>\n",
       "      <td>0.000701</td>\n",
       "      <td>0.001751</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.012652</td>\n",
       "      <td>0.001983</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.309323</td>\n",
       "      <td>0.000644</td>\n",
       "      <td>0.003026</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.070881</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.017306</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Parabacteroides_merdae</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.010123</td>\n",
       "      <td>0.011737</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.055457</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.241292</td>\n",
       "      <td>0.001944</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 1836 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                      taxa    G63225    G63227    G63228    G63229    G63230  \\\n",
       "0             UNCLASSIFIED  0.958397  0.413988  0.716716  0.968321  0.838273   \n",
       "1  Bifidobacterium_bifidum  0.000000  0.000000  0.001202  0.000000  0.000000   \n",
       "2  Akkermansia_muciniphila  0.000000  0.519963  0.205563  0.000000  0.000000   \n",
       "3      Ruminococcus_gnavus  0.000701  0.001751  0.000000  0.012652  0.001983   \n",
       "4   Parabacteroides_merdae  0.000000  0.010123  0.011737  0.000000  0.000000   \n",
       "\n",
       "     G63231    G63232    G63234    G63236  ...    G76485    G76486    G76487  \\\n",
       "0  0.950505  0.680969  0.944478  0.696196  ...  0.929316  0.910137  0.896830   \n",
       "1  0.000312  0.000000  0.000000  0.000000  ...  0.049548  0.015803  0.000000   \n",
       "2  0.000000  0.000000  0.006350  0.272486  ...  0.000000  0.000000  0.000000   \n",
       "3  0.000000  0.309323  0.000644  0.003026  ...  0.000000  0.070881  0.000000   \n",
       "4  0.000000  0.000000  0.000000  0.000000  ...  0.000000  0.000000  0.055457   \n",
       "\n",
       "     G76488    G76489    G76490    G76491    G76492    G76493  G76494  \n",
       "0  0.905858  0.740300  0.980560  0.830749  0.993131  0.954571       1  \n",
       "1  0.000000  0.000000  0.011534  0.000000  0.006869  0.045429       0  \n",
       "2  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000       0  \n",
       "3  0.000000  0.017306  0.000000  0.000000  0.000000  0.000000       0  \n",
       "4  0.000000  0.241292  0.001944  0.000000  0.000000  0.000000       0  \n",
       "\n",
       "[5 rows x 1836 columns]"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "amp_species.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### looking for the intersection of taxonomy at genus level\n",
    "#### calculate number of taxa in both vs. either one"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "# calculate number of taxa in both vs. either one"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "amplicon_genera_list = set(amp_genus[\"taxa\"])\n",
    "mgx_genera_list = set(mgx_genus[\"taxa\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [],
   "source": [
    "intersection_genera = amplicon_genera_list.intersection(mgx_genera_list)\n",
    "union_genera = amplicon_genera_list.union(mgx_genera_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "151"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/plain": [
       "151"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/plain": [
       "47"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(intersection_genera)\n",
    "len(amplicon_genera_list)-len(intersection_genera)\n",
    "len(mgx_genera_list)-len(intersection_genera)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [],
   "source": [
    "amplicon_genera = amplicon_genera_list.difference(intersection_genera)\n",
    "mgx_genera = mgx_genera_list.difference(intersection_genera)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "151"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/plain": [
       "47"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(amplicon_genera)\n",
    "len(mgx_genera)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### looking for the intersection of taxonomy at family level"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "63"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/plain": [
       "68"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/plain": [
       "9"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "amplicon_family_list = set(amp_family[\"taxa\"])\n",
    "mgx_family_list = set(mgx_family[\"taxname\"])\n",
    "\n",
    "intersection_family = amplicon_family_list.intersection(mgx_family_list)\n",
    "union_family = amplicon_family_list.union(mgx_family_list)\n",
    "\n",
    "len(intersection_family)\n",
    "len(amplicon_family_list)-len(intersection_family)\n",
    "len(mgx_family_list)-len(intersection_family)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "68"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/plain": [
       "9"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "amplicon_family = amplicon_family_list.difference(intersection_family)\n",
    "mgx_family = mgx_family_list.difference(intersection_family)\n",
    "\n",
    "len(amplicon_family)\n",
    "len(mgx_family)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### looking for the intersection of taxonomy at species level"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>taxname</th>\n",
       "      <th>G69146</th>\n",
       "      <th>G69147</th>\n",
       "      <th>G69148</th>\n",
       "      <th>G69149</th>\n",
       "      <th>G69150</th>\n",
       "      <th>G69151</th>\n",
       "      <th>G69152</th>\n",
       "      <th>G69153</th>\n",
       "      <th>G69154</th>\n",
       "      <th>...</th>\n",
       "      <th>G80617</th>\n",
       "      <th>G80619</th>\n",
       "      <th>G80620</th>\n",
       "      <th>G80621</th>\n",
       "      <th>G80623</th>\n",
       "      <th>G80624</th>\n",
       "      <th>G86980</th>\n",
       "      <th>G86986</th>\n",
       "      <th>G90143</th>\n",
       "      <th>G90147</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Escherichia_coli</td>\n",
       "      <td>0.284918</td>\n",
       "      <td>0.32781</td>\n",
       "      <td>0.004385</td>\n",
       "      <td>0.045543</td>\n",
       "      <td>0.020078</td>\n",
       "      <td>0.042352</td>\n",
       "      <td>0.050934</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.003681</td>\n",
       "      <td>...</td>\n",
       "      <td>0.196305</td>\n",
       "      <td>0.000056</td>\n",
       "      <td>0.036164</td>\n",
       "      <td>0.617818</td>\n",
       "      <td>0.000040</td>\n",
       "      <td>0.000136</td>\n",
       "      <td>0.319663</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.006707</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Akkermansia_muciniphila</td>\n",
       "      <td>0.242453</td>\n",
       "      <td>0.00563</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.079431</td>\n",
       "      <td>0.052597</td>\n",
       "      <td>0.003702</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000022</td>\n",
       "      <td>0.883796</td>\n",
       "      <td>0.186946</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002018</td>\n",
       "      <td>0.573473</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Klebsiella_pneumoniae</td>\n",
       "      <td>0.133182</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001958</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.132023</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.015799</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Dialister_invisus</td>\n",
       "      <td>0.057785</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.015439</td>\n",
       "      <td>0.000078</td>\n",
       "      <td>0.002291</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000010</td>\n",
       "      <td>0.032804</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001019</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Bifidobacterium_longum</td>\n",
       "      <td>0.041493</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.012710</td>\n",
       "      <td>0.119270</td>\n",
       "      <td>0.005591</td>\n",
       "      <td>0.171838</td>\n",
       "      <td>0.186589</td>\n",
       "      <td>0.009908</td>\n",
       "      <td>0.050695</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.011101</td>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.000713</td>\n",
       "      <td>0.167309</td>\n",
       "      <td>0.006347</td>\n",
       "      <td>0.056740</td>\n",
       "      <td>0.344709</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.021620</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 801 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   taxname    G69146   G69147    G69148    G69149    G69150  \\\n",
       "0         Escherichia_coli  0.284918  0.32781  0.004385  0.045543  0.020078   \n",
       "1  Akkermansia_muciniphila  0.242453  0.00563  0.000000  0.000000  0.000000   \n",
       "2    Klebsiella_pneumoniae  0.133182  0.00000  0.000000  0.000000  0.000000   \n",
       "3        Dialister_invisus  0.057785  0.00000  0.015439  0.000078  0.002291   \n",
       "4   Bifidobacterium_longum  0.041493  0.00000  0.012710  0.119270  0.005591   \n",
       "\n",
       "     G69151    G69152    G69153    G69154  ...    G80617    G80619    G80620  \\\n",
       "0  0.042352  0.050934  0.000000  0.003681  ...  0.196305  0.000056  0.036164   \n",
       "1  0.000000  0.079431  0.052597  0.003702  ...  0.000000  0.000000  0.000000   \n",
       "2  0.000000  0.000000  0.000000  0.000000  ...  0.000000  0.000000  0.000000   \n",
       "3  0.000000  0.000000  0.000000  0.000000  ...  0.000000  0.000000  0.000000   \n",
       "4  0.171838  0.186589  0.009908  0.050695  ...  0.000000  0.011101  0.000200   \n",
       "\n",
       "     G80621    G80623    G80624    G86980    G86986    G90143    G90147  \n",
       "0  0.617818  0.000040  0.000136  0.319663  0.000000  0.006707  0.000000  \n",
       "1  0.000000  0.000022  0.883796  0.186946  0.000000  0.002018  0.573473  \n",
       "2  0.001958  0.000000  0.000000  0.132023  0.000000  0.000000  0.015799  \n",
       "3  0.000000  0.000000  0.000010  0.032804  0.000000  0.001019  0.000000  \n",
       "4  0.000713  0.167309  0.006347  0.056740  0.344709  0.000000  0.021620  \n",
       "\n",
       "[5 rows x 801 columns]"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mgx_species.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>taxa</th>\n",
       "      <th>G63225</th>\n",
       "      <th>G63227</th>\n",
       "      <th>G63228</th>\n",
       "      <th>G63229</th>\n",
       "      <th>G63230</th>\n",
       "      <th>G63231</th>\n",
       "      <th>G63232</th>\n",
       "      <th>G63234</th>\n",
       "      <th>G63236</th>\n",
       "      <th>...</th>\n",
       "      <th>G76485</th>\n",
       "      <th>G76486</th>\n",
       "      <th>G76487</th>\n",
       "      <th>G76488</th>\n",
       "      <th>G76489</th>\n",
       "      <th>G76490</th>\n",
       "      <th>G76491</th>\n",
       "      <th>G76492</th>\n",
       "      <th>G76493</th>\n",
       "      <th>G76494</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>UNCLASSIFIED</td>\n",
       "      <td>0.958397</td>\n",
       "      <td>0.413988</td>\n",
       "      <td>0.716716</td>\n",
       "      <td>0.968321</td>\n",
       "      <td>0.838273</td>\n",
       "      <td>0.950505</td>\n",
       "      <td>0.680969</td>\n",
       "      <td>0.944478</td>\n",
       "      <td>0.696196</td>\n",
       "      <td>...</td>\n",
       "      <td>0.929316</td>\n",
       "      <td>0.910137</td>\n",
       "      <td>0.896830</td>\n",
       "      <td>0.905858</td>\n",
       "      <td>0.740300</td>\n",
       "      <td>0.980560</td>\n",
       "      <td>0.830749</td>\n",
       "      <td>0.993131</td>\n",
       "      <td>0.954571</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Bifidobacterium_bifidum</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001202</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000312</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.049548</td>\n",
       "      <td>0.015803</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.011534</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.006869</td>\n",
       "      <td>0.045429</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Akkermansia_muciniphila</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.519963</td>\n",
       "      <td>0.205563</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.006350</td>\n",
       "      <td>0.272486</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Ruminococcus_gnavus</td>\n",
       "      <td>0.000701</td>\n",
       "      <td>0.001751</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.012652</td>\n",
       "      <td>0.001983</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.309323</td>\n",
       "      <td>0.000644</td>\n",
       "      <td>0.003026</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.070881</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.017306</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Parabacteroides_merdae</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.010123</td>\n",
       "      <td>0.011737</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.055457</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.241292</td>\n",
       "      <td>0.001944</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 1836 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                      taxa    G63225    G63227    G63228    G63229    G63230  \\\n",
       "0             UNCLASSIFIED  0.958397  0.413988  0.716716  0.968321  0.838273   \n",
       "1  Bifidobacterium_bifidum  0.000000  0.000000  0.001202  0.000000  0.000000   \n",
       "2  Akkermansia_muciniphila  0.000000  0.519963  0.205563  0.000000  0.000000   \n",
       "3      Ruminococcus_gnavus  0.000701  0.001751  0.000000  0.012652  0.001983   \n",
       "4   Parabacteroides_merdae  0.000000  0.010123  0.011737  0.000000  0.000000   \n",
       "\n",
       "     G63231    G63232    G63234    G63236  ...    G76485    G76486    G76487  \\\n",
       "0  0.950505  0.680969  0.944478  0.696196  ...  0.929316  0.910137  0.896830   \n",
       "1  0.000312  0.000000  0.000000  0.000000  ...  0.049548  0.015803  0.000000   \n",
       "2  0.000000  0.000000  0.006350  0.272486  ...  0.000000  0.000000  0.000000   \n",
       "3  0.000000  0.309323  0.000644  0.003026  ...  0.000000  0.070881  0.000000   \n",
       "4  0.000000  0.000000  0.000000  0.000000  ...  0.000000  0.000000  0.055457   \n",
       "\n",
       "     G76488    G76489    G76490    G76491    G76492    G76493  G76494  \n",
       "0  0.905858  0.740300  0.980560  0.830749  0.993131  0.954571       1  \n",
       "1  0.000000  0.000000  0.011534  0.000000  0.006869  0.045429       0  \n",
       "2  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000       0  \n",
       "3  0.000000  0.017306  0.000000  0.000000  0.000000  0.000000       0  \n",
       "4  0.000000  0.241292  0.001944  0.000000  0.000000  0.000000       0  \n",
       "\n",
       "[5 rows x 1836 columns]"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "amp_species.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "127"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/plain": [
       "111"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/plain": [
       "384"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "amplicon_species_list = set(amp_species[\"taxa\"])\n",
    "mgx_species_list = set(mgx_species[\"taxname\"])\n",
    "\n",
    "intersection_species = amplicon_species_list.intersection(mgx_species_list)\n",
    "union_species = amplicon_species_list.union(mgx_species_list)\n",
    "\n",
    "len(intersection_species)\n",
    "len(amplicon_species_list)-len(intersection_species)\n",
    "len(mgx_species_list)-len(intersection_species)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "111"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/plain": [
       "384"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "amplicon_species = amplicon_species_list.difference(intersection_species)\n",
    "mgx_species = mgx_species_list.difference(intersection_species)\n",
    "\n",
    "len(amplicon_species)\n",
    "len(mgx_species)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### creating long-form dataframe with data from both sequencing methods"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>taxa</th>\n",
       "      <th>G69146</th>\n",
       "      <th>G69147</th>\n",
       "      <th>G69148</th>\n",
       "      <th>G69149</th>\n",
       "      <th>G69150</th>\n",
       "      <th>G69151</th>\n",
       "      <th>G69152</th>\n",
       "      <th>G69153</th>\n",
       "      <th>G69154</th>\n",
       "      <th>...</th>\n",
       "      <th>G80617</th>\n",
       "      <th>G80619</th>\n",
       "      <th>G80620</th>\n",
       "      <th>G80621</th>\n",
       "      <th>G80623</th>\n",
       "      <th>G80624</th>\n",
       "      <th>G86980</th>\n",
       "      <th>G86986</th>\n",
       "      <th>G90143</th>\n",
       "      <th>G90147</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Absiella</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Acidaminococcus</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.013550</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Acinetobacter</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Actinobaculum</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001091</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Actinomyces</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000045</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000083</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000766</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000011</td>\n",
       "      <td>0.000051</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 801 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              taxa  G69146  G69147  G69148  G69149  G69150  G69151    G69152  \\\n",
       "0         Absiella     0.0     0.0     0.0     0.0     0.0     0.0  0.000000   \n",
       "1  Acidaminococcus     0.0     0.0     0.0     0.0     0.0     0.0  0.000000   \n",
       "2    Acinetobacter     0.0     0.0     0.0     0.0     0.0     0.0  0.000000   \n",
       "3    Actinobaculum     0.0     0.0     0.0     0.0     0.0     0.0  0.000000   \n",
       "4      Actinomyces     0.0     0.0     0.0     0.0     0.0     0.0  0.000045   \n",
       "\n",
       "   G69153    G69154  ...  G80617  G80619  G80620  G80621  G80623    G80624  \\\n",
       "0     0.0  0.000000  ...     0.0     0.0     0.0     0.0     0.0  0.000000   \n",
       "1     0.0  0.013550  ...     0.0     0.0     0.0     0.0     0.0  0.000000   \n",
       "2     0.0  0.000000  ...     0.0     0.0     0.0     0.0     0.0  0.000000   \n",
       "3     0.0  0.000000  ...     0.0     0.0     0.0     0.0     0.0  0.000000   \n",
       "4     0.0  0.000083  ...     0.0     0.0     0.0     0.0     0.0  0.000766   \n",
       "\n",
       "   G86980  G86986    G90143    G90147  \n",
       "0     0.0     0.0  0.000000  0.000000  \n",
       "1     0.0     0.0  0.000000  0.000000  \n",
       "2     0.0     0.0  0.000000  0.000000  \n",
       "3     0.0     0.0  0.000000  0.001091  \n",
       "4     0.0     0.0  0.000011  0.000051  \n",
       "\n",
       "[5 rows x 801 columns]"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mgx_genus.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>taxa</th>\n",
       "      <th>G78487</th>\n",
       "      <th>G78488</th>\n",
       "      <th>G78489</th>\n",
       "      <th>G78490</th>\n",
       "      <th>G78492</th>\n",
       "      <th>G78493</th>\n",
       "      <th>G78494</th>\n",
       "      <th>G78495</th>\n",
       "      <th>G78496</th>\n",
       "      <th>...</th>\n",
       "      <th>G80605</th>\n",
       "      <th>G80594</th>\n",
       "      <th>G80611</th>\n",
       "      <th>G80617</th>\n",
       "      <th>G80623</th>\n",
       "      <th>G80556</th>\n",
       "      <th>G80563</th>\n",
       "      <th>G80570</th>\n",
       "      <th>G80576</th>\n",
       "      <th>G80582</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Abiotrophia</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Acetanaerobacterium</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000187</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Acetobacter</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Achromobacter</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Acidaminococcus</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 842 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  taxa  G78487    G78488  G78489  G78490  G78492  G78493  \\\n",
       "0          Abiotrophia     0.0  0.000000     0.0     0.0     0.0     0.0   \n",
       "1  Acetanaerobacterium     0.0  0.000187     0.0     0.0     0.0     0.0   \n",
       "2          Acetobacter     0.0  0.000000     0.0     0.0     0.0     0.0   \n",
       "3        Achromobacter     0.0  0.000000     0.0     0.0     0.0     0.0   \n",
       "4      Acidaminococcus     0.0  0.000000     0.0     0.0     0.0     0.0   \n",
       "\n",
       "   G78494  G78495  G78496  ...  G80605  G80594  G80611  G80617  G80623  \\\n",
       "0     0.0     0.0     0.0  ...     0.0     0.0     0.0     0.0     0.0   \n",
       "1     0.0     0.0     0.0  ...     0.0     0.0     0.0     0.0     0.0   \n",
       "2     0.0     0.0     0.0  ...     0.0     0.0     0.0     0.0     0.0   \n",
       "3     0.0     0.0     0.0  ...     0.0     0.0     0.0     0.0     0.0   \n",
       "4     0.0     0.0     0.0  ...     0.0     0.0     0.0     0.0     0.0   \n",
       "\n",
       "   G80556  G80563  G80570  G80576  G80582  \n",
       "0     0.0     0.0     0.0     0.0     0.0  \n",
       "1     0.0     0.0     0.0     0.0     0.0  \n",
       "2     0.0     0.0     0.0     0.0     0.0  \n",
       "3     0.0     0.0     0.0     0.0     0.0  \n",
       "4     0.0     0.0     0.0     0.0     0.0  \n",
       "\n",
       "[5 rows x 842 columns]"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "amp_genus.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [],
   "source": [
    "# only keeping samples we have both mgx and 16S profiles for\n",
    "samples_intersect = set(mgx_genus.columns.values).intersection(set(amp_genus.columns.values))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [],
   "source": [
    "# remove mothers\n",
    "children_intersect = [id for id in samples_intersect if not id.startswith(\"M\")]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "842"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(amp_genus.columns.values)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['taxa', 'G69146', 'G69147', 'G69148', 'G69149', 'G69150', 'G69151',\n",
       "       'G69152', 'G69153', 'G69154', 'G69155', 'G69156', 'G69157',\n",
       "       'G69158', 'G69159', 'G69160', 'G69161', 'G69162', 'G69163',\n",
       "       'G69164', 'G69165', 'G69166', 'G69167', 'G69168', 'G69169',\n",
       "       'G69170', 'G69171', 'G69172', 'G69173', 'G69174', 'G69175',\n",
       "       'G69176', 'G69177', 'G69178', 'G69179', 'G69180', 'G69181',\n",
       "       'G69182', 'G69183', 'G69184', 'G69185', 'G69186', 'G69187',\n",
       "       'G69188', 'G69189', 'G69190', 'G69191', 'G69192', 'G69193',\n",
       "       'G69194', 'G69195', 'G69196', 'G69197', 'G69198', 'G69199',\n",
       "       'G69201', 'G69202', 'G69203', 'G69204', 'G69205', 'G69206',\n",
       "       'G69207', 'G69208', 'G69209', 'G69210', 'G69211', 'G69212',\n",
       "       'G69213', 'G69214', 'G69215', 'G69216', 'G69217', 'G69218',\n",
       "       'G69219', 'G69221', 'G69222', 'G69223', 'G69224', 'G69225',\n",
       "       'G69226', 'G69227', 'G69228', 'G69229', 'G69230', 'G69231',\n",
       "       'G69232', 'G69233', 'G69234', 'G69235', 'G69236', 'G69237',\n",
       "       'G69238', 'G69253', 'G69256', 'G69265', 'G69268', 'G69273',\n",
       "       'G69274', 'G69280', 'G69286', 'G69287', 'G69291', 'G69295',\n",
       "       'G78487', 'G78488', 'G78489', 'G78490', 'G78492', 'G78493',\n",
       "       'G78494', 'G78495', 'G78496', 'G78498', 'G78499', 'G78500',\n",
       "       'G78502', 'G78504', 'G78505', 'G78507', 'G78508', 'G78509',\n",
       "       'G78510', 'G78511', 'G78512', 'G78513', 'G78514', 'G78515',\n",
       "       'G78516', 'G78517', 'G78518', 'G78519', 'G78520', 'G78521',\n",
       "       'G78522', 'G78523', 'G78524', 'G78525', 'G78526', 'G78527',\n",
       "       'G78528', 'G78529', 'G78530', 'G78531', 'G78532', 'G78533',\n",
       "       'G78534', 'G78535', 'G78536', 'G78537', 'G78538', 'G78539',\n",
       "       'G78540', 'G78541', 'G78542', 'G78543', 'G78544', 'G78545',\n",
       "       'G78546', 'G78547', 'G78548', 'G78550', 'G78551', 'G78552',\n",
       "       'G78553', 'G78554', 'G78555', 'G78556', 'G78557', 'G78558',\n",
       "       'G78559', 'G78561', 'G78562', 'G78563', 'G78564', 'G78565',\n",
       "       'G78566', 'G78567', 'G78568', 'G78571', 'G78572', 'G78573',\n",
       "       'G78574', 'G78575', 'G78576', 'G78577', 'G78578', 'G78579',\n",
       "       'G78580', 'G78581', 'G78582', 'G78583', 'G78585', 'G78586',\n",
       "       'G78587', 'G78588', 'G78589', 'G78590', 'G78591', 'G78592',\n",
       "       'G78593', 'G78594', 'G78595', 'G78596', 'G78597', 'G78598',\n",
       "       'G78599', 'G78600', 'G78601', 'G78602', 'G78603', 'G78604',\n",
       "       'G78605', 'G78606', 'G78607', 'G78608', 'G78609', 'G78610',\n",
       "       'G78611', 'G78612', 'G78613', 'G78614', 'G78615', 'G78616',\n",
       "       'G78617', 'G78618', 'G78619', 'G78620', 'G78621', 'G78623',\n",
       "       'G78624', 'G78625', 'G78626', 'G78627', 'G78628', 'G78629',\n",
       "       'G78630', 'G78631', 'G78633', 'G78634', 'G78635', 'G78636',\n",
       "       'G78638', 'G78640', 'G78642', 'G78644', 'G78645', 'G78646',\n",
       "       'G78647', 'G78648', 'G78649', 'G78650', 'G78651', 'G78652',\n",
       "       'G78653', 'G78654', 'G78655', 'G78656', 'G78658', 'G78660',\n",
       "       'G78661', 'G78662', 'G78663', 'G78664', 'G78665', 'G78666',\n",
       "       'G78667', 'G78668', 'G78669', 'G78670', 'G78671', 'G78672',\n",
       "       'G78673', 'G78674', 'G78675', 'G78676', 'G78677', 'G78679',\n",
       "       'G78680', 'G78681', 'G78682', 'G78683', 'G78684', 'G78685',\n",
       "       'G78686', 'G78687', 'G78688', 'G78690', 'G78692', 'G78693',\n",
       "       'G78694', 'G78696', 'G78697', 'G78699', 'G78701', 'G78704',\n",
       "       'G78706', 'G78707', 'G78708', 'G78712', 'G78713', 'G78714',\n",
       "       'G78715', 'G78716', 'G78717', 'G78718', 'G78719', 'G78720',\n",
       "       'G78721', 'G78722', 'G78727', 'G78728', 'G78730', 'G78732',\n",
       "       'G78733', 'G78734', 'G78735', 'G78737', 'G78738', 'G78739',\n",
       "       'G78740', 'G78742', 'G78743', 'G78744', 'G78746', 'G78747',\n",
       "       'G78748', 'G78749', 'G78750', 'G78751', 'G78752', 'G78757',\n",
       "       'G78758', 'G78759', 'G78762', 'G78763', 'G78765', 'G78766',\n",
       "       'G78768', 'G78769', 'G78770', 'G78772', 'G78773', 'G78774',\n",
       "       'G78775', 'G78776', 'G78777', 'G78778', 'G78780', 'G78782',\n",
       "       'G78783', 'G78784', 'G78787', 'G78788', 'G78789', 'G78790',\n",
       "       'G78791', 'G78792', 'G78793', 'G78794', 'G78795', 'G78796',\n",
       "       'G78797', 'G78798', 'G78799', 'G78801', 'G78802', 'G78803',\n",
       "       'G78804', 'G78805', 'G78806', 'G78807', 'G78808', 'G78809',\n",
       "       'G78810', 'G78811', 'G78814', 'G78815', 'G78816', 'G78817',\n",
       "       'G78818', 'G78819', 'G78820', 'G78821', 'G78822', 'G78823',\n",
       "       'G78824', 'G78825', 'G78826', 'G78827', 'G78828', 'G78829',\n",
       "       'G78830', 'G78831', 'G78832', 'G78833', 'G78834', 'G78835',\n",
       "       'G78836', 'G78837', 'G78838', 'G78839', 'G78840', 'G78841',\n",
       "       'G78842', 'G78843', 'G78844', 'G78845', 'G78847', 'G78849',\n",
       "       'G78850', 'G78851', 'G78852', 'G78853', 'G78854', 'G78855',\n",
       "       'G78856', 'G78857', 'G78858', 'G78859', 'G78860', 'G78861',\n",
       "       'G78862', 'G78863', 'G78864', 'G78865', 'G78866', 'G78867',\n",
       "       'G78868', 'G78869', 'G78870', 'G78871', 'G78872', 'G78873',\n",
       "       'G78874', 'G78875', 'G78876', 'G78877', 'G78878', 'G78879',\n",
       "       'G78880', 'G78881', 'G78882', 'G78883', 'G78884', 'G78885',\n",
       "       'G78886', 'G78887', 'G78889', 'G78890', 'G78891', 'G78892',\n",
       "       'G78893', 'G78894', 'G78895', 'G78896', 'G78897', 'G78898',\n",
       "       'G78899', 'G78900', 'G78901', 'G78902', 'G78903', 'G78904',\n",
       "       'G78905', 'G78906', 'G78907', 'G78908', 'G78909', 'G78910',\n",
       "       'G78911', 'G78912', 'G78913', 'G78914', 'G80150', 'G80158',\n",
       "       'G80263', 'G80265', 'G80266', 'G80267', 'G80268', 'G80271',\n",
       "       'G80272', 'G80273', 'G80274', 'G80275', 'G80276', 'G80277',\n",
       "       'G80278', 'G80279', 'G80280', 'G80281', 'G80282', 'G80283',\n",
       "       'G80284', 'G80285', 'G80286', 'G80287', 'G80288', 'G80290',\n",
       "       'G80291', 'G80292', 'G80293', 'G80294', 'G80295', 'G80296',\n",
       "       'G80297', 'G80298', 'G80299', 'G80300', 'G80301', 'G80302',\n",
       "       'G80303', 'G80304', 'G80305', 'G80306', 'G80307', 'G80308',\n",
       "       'G80309', 'G80310', 'G80312', 'G80313', 'G80314', 'G80315',\n",
       "       'G80316', 'G80317', 'G80318', 'G80319', 'G80320', 'G80321',\n",
       "       'G80322', 'G80323', 'G80324', 'G80325', 'G80326', 'G80327',\n",
       "       'G80328', 'G80329', 'G80330', 'G80331', 'G80332', 'G80333',\n",
       "       'G80334', 'G80335', 'G80336', 'G80337', 'G80338', 'G80339',\n",
       "       'G80340', 'G80341', 'G80342', 'G80343', 'G80344', 'G80345',\n",
       "       'G80346', 'G80347', 'G80348', 'G80350', 'G80351', 'G80352',\n",
       "       'G80353', 'G80354', 'G80355', 'G80356', 'G80357', 'G80358',\n",
       "       'G80359', 'G80360', 'G80361', 'G80362', 'G80363', 'G80364',\n",
       "       'G80365', 'G80366', 'G80368', 'G80369', 'G80370', 'G80371',\n",
       "       'G80372', 'G80373', 'G80374', 'G80375', 'G80376', 'G80377',\n",
       "       'G80378', 'G80379', 'G80380', 'G80381', 'G80382', 'G80383',\n",
       "       'G80384', 'G80385', 'G80386', 'G80387', 'G80388', 'G80389',\n",
       "       'G80390', 'G80391', 'G80392', 'G80393', 'G80394', 'G80395',\n",
       "       'G80396', 'G80397', 'G80398', 'G80399', 'G80400', 'G80401',\n",
       "       'G80402', 'G80403', 'G80404', 'G80405', 'G80406', 'G80407',\n",
       "       'G80408', 'G80409', 'G80410', 'G80411', 'G80412', 'G80413',\n",
       "       'G80414', 'G80415', 'G80416', 'G80417', 'G80418', 'G80419',\n",
       "       'G80420', 'G80421', 'G80422', 'G80423', 'G80425', 'G80426',\n",
       "       'G80427', 'G80428', 'G80430', 'G80431', 'G80432', 'G80433',\n",
       "       'G80434', 'G80435', 'G80436', 'G80437', 'G80438', 'G80439',\n",
       "       'G80440', 'G80441', 'G80444', 'G80446', 'G80447', 'G80448',\n",
       "       'G80449', 'G80450', 'G80451', 'G80452', 'G80454', 'G80455',\n",
       "       'G80456', 'G80457', 'G80459', 'G80460', 'G80462', 'G80464',\n",
       "       'G80465', 'G80468', 'G80469', 'G80470', 'G80472', 'G80474',\n",
       "       'G80475', 'G80476', 'G80477', 'G80478', 'G80479', 'G80481',\n",
       "       'G80482', 'G80483', 'G80484', 'G80485', 'G80486', 'G80487',\n",
       "       'G80488', 'G80489', 'G80490', 'G80491', 'G80492', 'G80493',\n",
       "       'G80494', 'G80496', 'G80497', 'G80499', 'G80500', 'G80501',\n",
       "       'G80502', 'G80503', 'G80505', 'G80506', 'G80507', 'G80508',\n",
       "       'G80509', 'G80510', 'G80512', 'G80513', 'G80514', 'G80515',\n",
       "       'G80516', 'G80517', 'G80518', 'G80520', 'G80522', 'G80523',\n",
       "       'G80524', 'G80526', 'G80527', 'G80528', 'G80529', 'G80530',\n",
       "       'G80531', 'G80532', 'G80533', 'G80534', 'G80535', 'G80536',\n",
       "       'G80537', 'G80538', 'G80539', 'G80540', 'G80541', 'G80542',\n",
       "       'G80543', 'G80545', 'G80547', 'G80549', 'G80550', 'G80552',\n",
       "       'G80553', 'G80554', 'G80555', 'G80556', 'G80557', 'G80558',\n",
       "       'G80560', 'G80561', 'G80562', 'G80563', 'G80565', 'G80566',\n",
       "       'G80567', 'G80568', 'G80569', 'G80570', 'G80571', 'G80572',\n",
       "       'G80573', 'G80574', 'G80575', 'G80576', 'G80577', 'G80578',\n",
       "       'G80579', 'G80580', 'G80581', 'G80582', 'G80583', 'G80584',\n",
       "       'G80585', 'G80587', 'G80588', 'G80589', 'G80590', 'G80591',\n",
       "       'G80592', 'G80593', 'G80594', 'G80595', 'G80596', 'G80598',\n",
       "       'G80599', 'G80601', 'G80602', 'G80603', 'G80604', 'G80605',\n",
       "       'G80606', 'G80608', 'G80609', 'G80610', 'G80611', 'G80612',\n",
       "       'G80613', 'G80614', 'G80615', 'G80616', 'G80617', 'G80619',\n",
       "       'G80620', 'G80621', 'G80623', 'G80624', 'G86980', 'G86986',\n",
       "       'G90143', 'G90147'], dtype=object)"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mgx_genus.columns.values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'G69146',\n",
       " 'G69147',\n",
       " 'G69148',\n",
       " 'G69149',\n",
       " 'G69150',\n",
       " 'G69152',\n",
       " 'G69153',\n",
       " 'G69154',\n",
       " 'G69155',\n",
       " 'G69156',\n",
       " 'G69157',\n",
       " 'G69158',\n",
       " 'G69159',\n",
       " 'G69160',\n",
       " 'G69161',\n",
       " 'G69162',\n",
       " 'G69163',\n",
       " 'G69164',\n",
       " 'G69166',\n",
       " 'G69167',\n",
       " 'G69170',\n",
       " 'G69171',\n",
       " 'G69172',\n",
       " 'G69173',\n",
       " 'G69174',\n",
       " 'G69175',\n",
       " 'G69176',\n",
       " 'G69177',\n",
       " 'G69178',\n",
       " 'G69179',\n",
       " 'G69180',\n",
       " 'G69181',\n",
       " 'G69182',\n",
       " 'G69183',\n",
       " 'G69184',\n",
       " 'G69185',\n",
       " 'G69186',\n",
       " 'G69187',\n",
       " 'G69188',\n",
       " 'G69189',\n",
       " 'G69190',\n",
       " 'G69191',\n",
       " 'G69192',\n",
       " 'G69193',\n",
       " 'G69194',\n",
       " 'G69195',\n",
       " 'G69196',\n",
       " 'G69197',\n",
       " 'G69198',\n",
       " 'G69199',\n",
       " 'G69201',\n",
       " 'G69202',\n",
       " 'G69203',\n",
       " 'G69204',\n",
       " 'G69205',\n",
       " 'G69206',\n",
       " 'G69207',\n",
       " 'G69208',\n",
       " 'G69209',\n",
       " 'G69210',\n",
       " 'G69211',\n",
       " 'G69212',\n",
       " 'G69213',\n",
       " 'G69214',\n",
       " 'G69216',\n",
       " 'G69217',\n",
       " 'G69218',\n",
       " 'G69219',\n",
       " 'G69221',\n",
       " 'G69222',\n",
       " 'G69223',\n",
       " 'G69224',\n",
       " 'G69225',\n",
       " 'G69226',\n",
       " 'G69227',\n",
       " 'G69228',\n",
       " 'G69229',\n",
       " 'G69230',\n",
       " 'G69231',\n",
       " 'G69232',\n",
       " 'G69233',\n",
       " 'G69234',\n",
       " 'G69235',\n",
       " 'G69236',\n",
       " 'G69238',\n",
       " 'G78487',\n",
       " 'G78488',\n",
       " 'G78489',\n",
       " 'G78490',\n",
       " 'G78492',\n",
       " 'G78493',\n",
       " 'G78494',\n",
       " 'G78495',\n",
       " 'G78496',\n",
       " 'G78498',\n",
       " 'G78499',\n",
       " 'G78500',\n",
       " 'G78502',\n",
       " 'G78504',\n",
       " 'G78505',\n",
       " 'G78507',\n",
       " 'G78508',\n",
       " 'G78509',\n",
       " 'G78510',\n",
       " 'G78511',\n",
       " 'G78512',\n",
       " 'G78513',\n",
       " 'G78514',\n",
       " 'G78515',\n",
       " 'G78516',\n",
       " 'G78517',\n",
       " 'G78518',\n",
       " 'G78519',\n",
       " 'G78520',\n",
       " 'G78521',\n",
       " 'G78522',\n",
       " 'G78523',\n",
       " 'G78524',\n",
       " 'G78525',\n",
       " 'G78526',\n",
       " 'G78527',\n",
       " 'G78528',\n",
       " 'G78529',\n",
       " 'G78530',\n",
       " 'G78531',\n",
       " 'G78532',\n",
       " 'G78533',\n",
       " 'G78534',\n",
       " 'G78535',\n",
       " 'G78536',\n",
       " 'G78537',\n",
       " 'G78538',\n",
       " 'G78539',\n",
       " 'G78540',\n",
       " 'G78541',\n",
       " 'G78542',\n",
       " 'G78543',\n",
       " 'G78544',\n",
       " 'G78545',\n",
       " 'G78546',\n",
       " 'G78547',\n",
       " 'G78548',\n",
       " 'G78550',\n",
       " 'G78551',\n",
       " 'G78552',\n",
       " 'G78553',\n",
       " 'G78554',\n",
       " 'G78555',\n",
       " 'G78556',\n",
       " 'G78557',\n",
       " 'G78558',\n",
       " 'G78559',\n",
       " 'G78561',\n",
       " 'G78562',\n",
       " 'G78563',\n",
       " 'G78564',\n",
       " 'G78565',\n",
       " 'G78566',\n",
       " 'G78567',\n",
       " 'G78568',\n",
       " 'G78571',\n",
       " 'G78572',\n",
       " 'G78573',\n",
       " 'G78574',\n",
       " 'G78575',\n",
       " 'G78576',\n",
       " 'G78577',\n",
       " 'G78578',\n",
       " 'G78579',\n",
       " 'G78580',\n",
       " 'G78581',\n",
       " 'G78582',\n",
       " 'G78583',\n",
       " 'G78585',\n",
       " 'G78586',\n",
       " 'G78587',\n",
       " 'G78588',\n",
       " 'G78589',\n",
       " 'G78590',\n",
       " 'G78591',\n",
       " 'G78592',\n",
       " 'G78593',\n",
       " 'G78594',\n",
       " 'G78595',\n",
       " 'G78596',\n",
       " 'G78597',\n",
       " 'G78598',\n",
       " 'G78599',\n",
       " 'G78600',\n",
       " 'G78601',\n",
       " 'G78602',\n",
       " 'G78603',\n",
       " 'G78604',\n",
       " 'G78605',\n",
       " 'G78606',\n",
       " 'G78607',\n",
       " 'G78608',\n",
       " 'G78609',\n",
       " 'G78610',\n",
       " 'G78611',\n",
       " 'G78612',\n",
       " 'G78613',\n",
       " 'G78614',\n",
       " 'G78615',\n",
       " 'G78616',\n",
       " 'G78617',\n",
       " 'G78618',\n",
       " 'G78619',\n",
       " 'G78620',\n",
       " 'G78621',\n",
       " 'G78623',\n",
       " 'G78624',\n",
       " 'G78625',\n",
       " 'G78626',\n",
       " 'G78627',\n",
       " 'G78628',\n",
       " 'G78629',\n",
       " 'G78630',\n",
       " 'G78631',\n",
       " 'G78633',\n",
       " 'G78634',\n",
       " 'G78635',\n",
       " 'G78636',\n",
       " 'G78638',\n",
       " 'G78640',\n",
       " 'G78642',\n",
       " 'G78644',\n",
       " 'G78645',\n",
       " 'G78646',\n",
       " 'G78647',\n",
       " 'G78648',\n",
       " 'G78649',\n",
       " 'G78650',\n",
       " 'G78651',\n",
       " 'G78652',\n",
       " 'G78653',\n",
       " 'G78654',\n",
       " 'G78655',\n",
       " 'G78656',\n",
       " 'G78658',\n",
       " 'G78660',\n",
       " 'G78661',\n",
       " 'G78662',\n",
       " 'G78663',\n",
       " 'G78664',\n",
       " 'G78665',\n",
       " 'G78666',\n",
       " 'G78667',\n",
       " 'G78668',\n",
       " 'G78669',\n",
       " 'G78670',\n",
       " 'G78671',\n",
       " 'G78672',\n",
       " 'G78673',\n",
       " 'G78674',\n",
       " 'G78675',\n",
       " 'G78676',\n",
       " 'G78677',\n",
       " 'G78679',\n",
       " 'G78680',\n",
       " 'G78681',\n",
       " 'G78682',\n",
       " 'G78683',\n",
       " 'G78684',\n",
       " 'G78685',\n",
       " 'G78686',\n",
       " 'G78687',\n",
       " 'G78688',\n",
       " 'G78690',\n",
       " 'G78692',\n",
       " 'G78693',\n",
       " 'G78694',\n",
       " 'G78696',\n",
       " 'G78697',\n",
       " 'G78699',\n",
       " 'G78701',\n",
       " 'G78704',\n",
       " 'G78706',\n",
       " 'G78707',\n",
       " 'G78708',\n",
       " 'G78712',\n",
       " 'G78713',\n",
       " 'G78714',\n",
       " 'G78715',\n",
       " 'G78716',\n",
       " 'G78717',\n",
       " 'G78718',\n",
       " 'G78719',\n",
       " 'G78720',\n",
       " 'G78721',\n",
       " 'G78722',\n",
       " 'G78727',\n",
       " 'G78728',\n",
       " 'G78730',\n",
       " 'G78732',\n",
       " 'G78733',\n",
       " 'G78734',\n",
       " 'G78735',\n",
       " 'G78737',\n",
       " 'G78738',\n",
       " 'G78739',\n",
       " 'G78740',\n",
       " 'G78742',\n",
       " 'G78743',\n",
       " 'G78744',\n",
       " 'G78746',\n",
       " 'G78747',\n",
       " 'G78748',\n",
       " 'G78749',\n",
       " 'G78750',\n",
       " 'G78751',\n",
       " 'G78752',\n",
       " 'G78757',\n",
       " 'G78758',\n",
       " 'G78762',\n",
       " 'G78763',\n",
       " 'G78765',\n",
       " 'G78766',\n",
       " 'G78768',\n",
       " 'G78769',\n",
       " 'G78770',\n",
       " 'G78772',\n",
       " 'G78773',\n",
       " 'G78774',\n",
       " 'G78775',\n",
       " 'G78776',\n",
       " 'G78777',\n",
       " 'G78778',\n",
       " 'G78780',\n",
       " 'G78782',\n",
       " 'G78783',\n",
       " 'G78784',\n",
       " 'G78787',\n",
       " 'G78788',\n",
       " 'G78789',\n",
       " 'G78790',\n",
       " 'G78791',\n",
       " 'G78792',\n",
       " 'G78793',\n",
       " 'G78794',\n",
       " 'G78795',\n",
       " 'G78796',\n",
       " 'G78797',\n",
       " 'G78798',\n",
       " 'G78799',\n",
       " 'G78801',\n",
       " 'G78802',\n",
       " 'G78803',\n",
       " 'G78804',\n",
       " 'G78805',\n",
       " 'G78806',\n",
       " 'G78807',\n",
       " 'G78808',\n",
       " 'G78809',\n",
       " 'G78811',\n",
       " 'G78814',\n",
       " 'G78815',\n",
       " 'G78816',\n",
       " 'G78817',\n",
       " 'G78818',\n",
       " 'G78819',\n",
       " 'G78820',\n",
       " 'G78821',\n",
       " 'G78823',\n",
       " 'G78824',\n",
       " 'G78825',\n",
       " 'G78826',\n",
       " 'G78827',\n",
       " 'G78828',\n",
       " 'G78829',\n",
       " 'G78830',\n",
       " 'G78831',\n",
       " 'G78832',\n",
       " 'G78833',\n",
       " 'G78834',\n",
       " 'G78835',\n",
       " 'G78836',\n",
       " 'G78837',\n",
       " 'G78838',\n",
       " 'G78839',\n",
       " 'G78841',\n",
       " 'G78842',\n",
       " 'G78843',\n",
       " 'G78844',\n",
       " 'G78845',\n",
       " 'G78847',\n",
       " 'G78849',\n",
       " 'G78850',\n",
       " 'G78851',\n",
       " 'G78852',\n",
       " 'G78853',\n",
       " 'G78854',\n",
       " 'G78855',\n",
       " 'G78856',\n",
       " 'G78857',\n",
       " 'G78858',\n",
       " 'G78859',\n",
       " 'G78860',\n",
       " 'G78861',\n",
       " 'G78862',\n",
       " 'G78863',\n",
       " 'G78864',\n",
       " 'G78865',\n",
       " 'G78866',\n",
       " 'G78867',\n",
       " 'G78868',\n",
       " 'G78869',\n",
       " 'G78870',\n",
       " 'G78871',\n",
       " 'G78872',\n",
       " 'G78873',\n",
       " 'G78874',\n",
       " 'G78875',\n",
       " 'G78876',\n",
       " 'G78877',\n",
       " 'G78878',\n",
       " 'G78879',\n",
       " 'G78880',\n",
       " 'G78881',\n",
       " 'G78882',\n",
       " 'G78883',\n",
       " 'G78884',\n",
       " 'G78885',\n",
       " 'G78886',\n",
       " 'G78887',\n",
       " 'G78889',\n",
       " 'G78890',\n",
       " 'G78891',\n",
       " 'G78892',\n",
       " 'G78893',\n",
       " 'G78894',\n",
       " 'G78895',\n",
       " 'G78896',\n",
       " 'G78897',\n",
       " 'G78898',\n",
       " 'G78899',\n",
       " 'G78900',\n",
       " 'G78901',\n",
       " 'G78902',\n",
       " 'G78903',\n",
       " 'G78904',\n",
       " 'G78905',\n",
       " 'G78906',\n",
       " 'G78907',\n",
       " 'G78908',\n",
       " 'G78909',\n",
       " 'G78910',\n",
       " 'G78911',\n",
       " 'G78912',\n",
       " 'G78913',\n",
       " 'G78914',\n",
       " 'G80150',\n",
       " 'G80158',\n",
       " 'G80263',\n",
       " 'G80265',\n",
       " 'G80266',\n",
       " 'G80267',\n",
       " 'G80268',\n",
       " 'G80271',\n",
       " 'G80272',\n",
       " 'G80273',\n",
       " 'G80274',\n",
       " 'G80275',\n",
       " 'G80276',\n",
       " 'G80277',\n",
       " 'G80278',\n",
       " 'G80279',\n",
       " 'G80280',\n",
       " 'G80281',\n",
       " 'G80282',\n",
       " 'G80283',\n",
       " 'G80284',\n",
       " 'G80285',\n",
       " 'G80286',\n",
       " 'G80287',\n",
       " 'G80288',\n",
       " 'G80290',\n",
       " 'G80291',\n",
       " 'G80292',\n",
       " 'G80293',\n",
       " 'G80294',\n",
       " 'G80295',\n",
       " 'G80296',\n",
       " 'G80297',\n",
       " 'G80298',\n",
       " 'G80299',\n",
       " 'G80300',\n",
       " 'G80301',\n",
       " 'G80302',\n",
       " 'G80303',\n",
       " 'G80304',\n",
       " 'G80305',\n",
       " 'G80306',\n",
       " 'G80307',\n",
       " 'G80308',\n",
       " 'G80309',\n",
       " 'G80310',\n",
       " 'G80312',\n",
       " 'G80313',\n",
       " 'G80314',\n",
       " 'G80315',\n",
       " 'G80316',\n",
       " 'G80317',\n",
       " 'G80318',\n",
       " 'G80319',\n",
       " 'G80320',\n",
       " 'G80321',\n",
       " 'G80322',\n",
       " 'G80323',\n",
       " 'G80324',\n",
       " 'G80325',\n",
       " 'G80326',\n",
       " 'G80327',\n",
       " 'G80328',\n",
       " 'G80329',\n",
       " 'G80330',\n",
       " 'G80331',\n",
       " 'G80332',\n",
       " 'G80333',\n",
       " 'G80334',\n",
       " 'G80335',\n",
       " 'G80336',\n",
       " 'G80337',\n",
       " 'G80338',\n",
       " 'G80339',\n",
       " 'G80340',\n",
       " 'G80341',\n",
       " 'G80342',\n",
       " 'G80343',\n",
       " 'G80344',\n",
       " 'G80345',\n",
       " 'G80346',\n",
       " 'G80347',\n",
       " 'G80348',\n",
       " 'G80350',\n",
       " 'G80351',\n",
       " 'G80352',\n",
       " 'G80353',\n",
       " 'G80354',\n",
       " 'G80355',\n",
       " 'G80356',\n",
       " 'G80357',\n",
       " 'G80358',\n",
       " 'G80359',\n",
       " 'G80360',\n",
       " 'G80361',\n",
       " 'G80362',\n",
       " 'G80363',\n",
       " 'G80364',\n",
       " 'G80365',\n",
       " 'G80366',\n",
       " 'G80368',\n",
       " 'G80369',\n",
       " 'G80370',\n",
       " 'G80371',\n",
       " 'G80372',\n",
       " 'G80373',\n",
       " 'G80374',\n",
       " 'G80375',\n",
       " 'G80376',\n",
       " 'G80377',\n",
       " 'G80378',\n",
       " 'G80379',\n",
       " 'G80380',\n",
       " 'G80381',\n",
       " 'G80382',\n",
       " 'G80383',\n",
       " 'G80384',\n",
       " 'G80385',\n",
       " 'G80386',\n",
       " 'G80387',\n",
       " 'G80388',\n",
       " 'G80389',\n",
       " 'G80390',\n",
       " 'G80391',\n",
       " 'G80392',\n",
       " 'G80393',\n",
       " 'G80394',\n",
       " 'G80395',\n",
       " 'G80396',\n",
       " 'G80397',\n",
       " 'G80398',\n",
       " 'G80399',\n",
       " 'G80400',\n",
       " 'G80401',\n",
       " 'G80402',\n",
       " 'G80403',\n",
       " 'G80404',\n",
       " 'G80405',\n",
       " 'G80406',\n",
       " 'G80407',\n",
       " 'G80408',\n",
       " 'G80409',\n",
       " 'G80410',\n",
       " 'G80411',\n",
       " 'G80412',\n",
       " 'G80413',\n",
       " 'G80414',\n",
       " 'G80415',\n",
       " 'G80416',\n",
       " 'G80417',\n",
       " 'G80418',\n",
       " 'G80419',\n",
       " 'G80420',\n",
       " 'G80421',\n",
       " 'G80422',\n",
       " 'G80423',\n",
       " 'G80425',\n",
       " 'G80426',\n",
       " 'G80427',\n",
       " 'G80428',\n",
       " 'G80430',\n",
       " 'G80431',\n",
       " 'G80432',\n",
       " 'G80433',\n",
       " 'G80434',\n",
       " 'G80435',\n",
       " 'G80436',\n",
       " 'G80437',\n",
       " 'G80438',\n",
       " 'G80439',\n",
       " 'G80440',\n",
       " 'G80441',\n",
       " 'G80444',\n",
       " 'G80446',\n",
       " 'G80447',\n",
       " 'G80448',\n",
       " 'G80449',\n",
       " 'G80450',\n",
       " 'G80451',\n",
       " 'G80452',\n",
       " 'G80454',\n",
       " 'G80455',\n",
       " 'G80456',\n",
       " 'G80457',\n",
       " 'G80459',\n",
       " 'G80460',\n",
       " 'G80462',\n",
       " 'G80464',\n",
       " 'G80465',\n",
       " 'G80468',\n",
       " 'G80469',\n",
       " 'G80470',\n",
       " 'G80472',\n",
       " 'G80474',\n",
       " 'G80475',\n",
       " 'G80476',\n",
       " 'G80477',\n",
       " 'G80478',\n",
       " 'G80479',\n",
       " 'G80481',\n",
       " 'G80482',\n",
       " 'G80483',\n",
       " 'G80484',\n",
       " 'G80485',\n",
       " 'G80486',\n",
       " 'G80487',\n",
       " 'G80488',\n",
       " 'G80489',\n",
       " 'G80490',\n",
       " 'G80491',\n",
       " 'G80492',\n",
       " 'G80493',\n",
       " 'G80494',\n",
       " 'G80496',\n",
       " 'G80497',\n",
       " 'G80499',\n",
       " 'G80500',\n",
       " 'G80501',\n",
       " 'G80502',\n",
       " 'G80503',\n",
       " 'G80505',\n",
       " 'G80506',\n",
       " 'G80507',\n",
       " 'G80508',\n",
       " 'G80509',\n",
       " 'G80510',\n",
       " 'G80512',\n",
       " 'G80513',\n",
       " 'G80514',\n",
       " 'G80515',\n",
       " 'G80516',\n",
       " 'G80517',\n",
       " 'G80518',\n",
       " 'G80520',\n",
       " 'G80522',\n",
       " 'G80523',\n",
       " 'G80524',\n",
       " 'G80526',\n",
       " 'G80527',\n",
       " 'G80528',\n",
       " 'G80529',\n",
       " 'G80530',\n",
       " 'G80531',\n",
       " 'G80532',\n",
       " 'G80533',\n",
       " 'G80534',\n",
       " 'G80535',\n",
       " 'G80536',\n",
       " 'G80537',\n",
       " 'G80538',\n",
       " 'G80539',\n",
       " 'G80540',\n",
       " 'G80541',\n",
       " 'G80542',\n",
       " 'G80543',\n",
       " 'G80545',\n",
       " 'G80547',\n",
       " 'G80549',\n",
       " 'G80550',\n",
       " 'G80552',\n",
       " 'G80553',\n",
       " 'G80554',\n",
       " 'G80555',\n",
       " 'G80556',\n",
       " 'G80557',\n",
       " 'G80558',\n",
       " 'G80560',\n",
       " 'G80561',\n",
       " 'G80562',\n",
       " 'G80563',\n",
       " 'G80565',\n",
       " 'G80566',\n",
       " 'G80567',\n",
       " 'G80568',\n",
       " 'G80569',\n",
       " 'G80570',\n",
       " 'G80571',\n",
       " 'G80572',\n",
       " 'G80573',\n",
       " 'G80574',\n",
       " 'G80575',\n",
       " 'G80576',\n",
       " 'G80577',\n",
       " 'G80578',\n",
       " 'G80579',\n",
       " 'G80580',\n",
       " 'G80581',\n",
       " 'G80582',\n",
       " 'G80583',\n",
       " 'G80584',\n",
       " 'G80585',\n",
       " 'G80587',\n",
       " 'G80588',\n",
       " 'G80589',\n",
       " 'G80590',\n",
       " 'G80591',\n",
       " 'G80592',\n",
       " 'G80593',\n",
       " 'G80594',\n",
       " 'G80595',\n",
       " 'G80596',\n",
       " 'G80598',\n",
       " 'G80599',\n",
       " 'G80601',\n",
       " 'G80602',\n",
       " 'G80603',\n",
       " 'G80604',\n",
       " 'G80605',\n",
       " 'G80606',\n",
       " 'G80608',\n",
       " 'G80609',\n",
       " 'G80610',\n",
       " 'G80611',\n",
       " 'G80612',\n",
       " 'G80613',\n",
       " 'G80614',\n",
       " 'G80615',\n",
       " 'G80616',\n",
       " 'G80617',\n",
       " 'G80619',\n",
       " 'G80620',\n",
       " 'G80621',\n",
       " 'G80623',\n",
       " 'G80624',\n",
       " 'taxa'}"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "samples_intersect"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "776"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(samples_intersect)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>subjectID</th>\n",
       "      <th>SampleID</th>\n",
       "      <th>age_at_collection</th>\n",
       "      <th>collection_month</th>\n",
       "      <th>delivery</th>\n",
       "      <th>gest_time</th>\n",
       "      <th>gender</th>\n",
       "      <th>country</th>\n",
       "      <th>Exclusive_breast_feeding</th>\n",
       "      <th>Breast_feeding_end</th>\n",
       "      <th>...</th>\n",
       "      <th>allergy_dog</th>\n",
       "      <th>allergy_birch</th>\n",
       "      <th>allergy_timothy</th>\n",
       "      <th>gid_wgs</th>\n",
       "      <th>mgx_reads</th>\n",
       "      <th>mgx_pool</th>\n",
       "      <th>mgx_reads_filtered</th>\n",
       "      <th>read_count_16S</th>\n",
       "      <th>sequencing_PDO_16S</th>\n",
       "      <th>gid_16s</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>E002338</td>\n",
       "      <td>3101193</td>\n",
       "      <td>304</td>\n",
       "      <td>10</td>\n",
       "      <td>vaginal</td>\n",
       "      <td>285</td>\n",
       "      <td>Female</td>\n",
       "      <td>FIN</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>...</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>G80541</td>\n",
       "      <td>28876314</td>\n",
       "      <td>Plate 7</td>\n",
       "      <td>28.568962</td>\n",
       "      <td>15870</td>\n",
       "      <td>PDO-4356</td>\n",
       "      <td>G75694</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>E002338</td>\n",
       "      <td>3107294</td>\n",
       "      <td>493</td>\n",
       "      <td>16</td>\n",
       "      <td>vaginal</td>\n",
       "      <td>285</td>\n",
       "      <td>Female</td>\n",
       "      <td>FIN</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>...</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>G80537</td>\n",
       "      <td>29615868</td>\n",
       "      <td>Plate 7</td>\n",
       "      <td>29.542272</td>\n",
       "      <td>13609</td>\n",
       "      <td>PDO-4356</td>\n",
       "      <td>G75855</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>E002338</td>\n",
       "      <td>3113022</td>\n",
       "      <td>852</td>\n",
       "      <td>28</td>\n",
       "      <td>vaginal</td>\n",
       "      <td>285</td>\n",
       "      <td>Female</td>\n",
       "      <td>FIN</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>...</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>G80322</td>\n",
       "      <td>30478184</td>\n",
       "      <td>Plate 8</td>\n",
       "      <td>30.41078</td>\n",
       "      <td>34620</td>\n",
       "      <td>PDO-4157</td>\n",
       "      <td>G73882</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>E002338</td>\n",
       "      <td>3107293</td>\n",
       "      <td>399</td>\n",
       "      <td>13</td>\n",
       "      <td>vaginal</td>\n",
       "      <td>285</td>\n",
       "      <td>Female</td>\n",
       "      <td>FIN</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>...</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>G80513</td>\n",
       "      <td>28503488</td>\n",
       "      <td>Plate 7</td>\n",
       "      <td>28.360762</td>\n",
       "      <td>7337</td>\n",
       "      <td>PDO-4356</td>\n",
       "      <td>G75849</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>E002338</td>\n",
       "      <td>3101190</td>\n",
       "      <td>212</td>\n",
       "      <td>7</td>\n",
       "      <td>vaginal</td>\n",
       "      <td>285</td>\n",
       "      <td>Female</td>\n",
       "      <td>FIN</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>...</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NaN</td>\n",
       "      <td>23928698</td>\n",
       "      <td>Diab Plate 9</td>\n",
       "      <td>23.824676</td>\n",
       "      <td>18931</td>\n",
       "      <td>PDO-4356</td>\n",
       "      <td>G75788</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 59 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Unnamed: 0 subjectID SampleID age_at_collection collection_month delivery  \\\n",
       "5            E002338  3101193               304               10  vaginal   \n",
       "6            E002338  3107294               493               16  vaginal   \n",
       "7            E002338  3113022               852               28  vaginal   \n",
       "8            E002338  3107293               399               13  vaginal   \n",
       "15           E002338  3101190               212                7  vaginal   \n",
       "\n",
       "Unnamed: 0 gest_time  gender country Exclusive_breast_feeding  \\\n",
       "5                285  Female     FIN                    FALSE   \n",
       "6                285  Female     FIN                    FALSE   \n",
       "7                285  Female     FIN                    FALSE   \n",
       "8                285  Female     FIN                    FALSE   \n",
       "15               285  Female     FIN                    FALSE   \n",
       "\n",
       "Unnamed: 0 Breast_feeding_end  ... allergy_dog allergy_birch allergy_timothy  \\\n",
       "5                       FALSE  ...       FALSE         FALSE           FALSE   \n",
       "6                        TRUE  ...       FALSE         FALSE           FALSE   \n",
       "7                        TRUE  ...       FALSE         FALSE           FALSE   \n",
       "8                       FALSE  ...       FALSE         FALSE           FALSE   \n",
       "15                      FALSE  ...       FALSE         FALSE           FALSE   \n",
       "\n",
       "Unnamed: 0 gid_wgs mgx_reads      mgx_pool mgx_reads_filtered read_count_16S  \\\n",
       "5           G80541  28876314       Plate 7          28.568962          15870   \n",
       "6           G80537  29615868       Plate 7          29.542272          13609   \n",
       "7           G80322  30478184       Plate 8           30.41078          34620   \n",
       "8           G80513  28503488       Plate 7          28.360762           7337   \n",
       "15             NaN  23928698  Diab Plate 9          23.824676          18931   \n",
       "\n",
       "Unnamed: 0 sequencing_PDO_16S gid_16s  \n",
       "5                    PDO-4356  G75694  \n",
       "6                    PDO-4356  G75855  \n",
       "7                    PDO-4157  G73882  \n",
       "8                    PDO-4356  G75849  \n",
       "15                   PDO-4356  G75788  \n",
       "\n",
       "[5 rows x 59 columns]"
      ]
     },
     "execution_count": 93,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metadata.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [],
   "source": [
    "metadata_filtered = metadata[metadata['gid_wgs'].isin(samples_intersect)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/danielle/miniconda2/lib/python3.7/site-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \"\"\"Entry point for launching an IPython kernel.\n"
     ]
    }
   ],
   "source": [
    "metadata_filtered[\"age_months\"] = pd.to_numeric(metadata_filtered[\"age_at_collection\"])/30"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {},
   "outputs": [],
   "source": [
    "def apply_dev_stage(row):\n",
    "    if row < 15:\n",
    "        return \"less than 15 months\"\n",
    "    elif row <= 30:\n",
    "        return \"15 to 30 months\"\n",
    "    else:\n",
    "        return \"older than 30 months\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/danielle/miniconda2/lib/python3.7/site-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \"\"\"Entry point for launching an IPython kernel.\n"
     ]
    }
   ],
   "source": [
    "metadata_filtered['dev_stage'] = metadata_filtered[\"age_months\"].apply(apply_dev_stage)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>subjectID</th>\n",
       "      <th>SampleID</th>\n",
       "      <th>age_at_collection</th>\n",
       "      <th>collection_month</th>\n",
       "      <th>delivery</th>\n",
       "      <th>gest_time</th>\n",
       "      <th>gender</th>\n",
       "      <th>country</th>\n",
       "      <th>Exclusive_breast_feeding</th>\n",
       "      <th>Breast_feeding_end</th>\n",
       "      <th>...</th>\n",
       "      <th>allergy_timothy</th>\n",
       "      <th>gid_wgs</th>\n",
       "      <th>mgx_reads</th>\n",
       "      <th>mgx_pool</th>\n",
       "      <th>mgx_reads_filtered</th>\n",
       "      <th>read_count_16S</th>\n",
       "      <th>sequencing_PDO_16S</th>\n",
       "      <th>gid_16s</th>\n",
       "      <th>age_months</th>\n",
       "      <th>dev_stage</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>E002338</td>\n",
       "      <td>3101193</td>\n",
       "      <td>304</td>\n",
       "      <td>10</td>\n",
       "      <td>vaginal</td>\n",
       "      <td>285</td>\n",
       "      <td>Female</td>\n",
       "      <td>FIN</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>...</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>G80541</td>\n",
       "      <td>28876314</td>\n",
       "      <td>Plate 7</td>\n",
       "      <td>28.568962</td>\n",
       "      <td>15870</td>\n",
       "      <td>PDO-4356</td>\n",
       "      <td>G75694</td>\n",
       "      <td>10.133333</td>\n",
       "      <td>less than 15 months</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>E002338</td>\n",
       "      <td>3107294</td>\n",
       "      <td>493</td>\n",
       "      <td>16</td>\n",
       "      <td>vaginal</td>\n",
       "      <td>285</td>\n",
       "      <td>Female</td>\n",
       "      <td>FIN</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>...</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>G80537</td>\n",
       "      <td>29615868</td>\n",
       "      <td>Plate 7</td>\n",
       "      <td>29.542272</td>\n",
       "      <td>13609</td>\n",
       "      <td>PDO-4356</td>\n",
       "      <td>G75855</td>\n",
       "      <td>16.433333</td>\n",
       "      <td>15 to 30 months</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>E002338</td>\n",
       "      <td>3113022</td>\n",
       "      <td>852</td>\n",
       "      <td>28</td>\n",
       "      <td>vaginal</td>\n",
       "      <td>285</td>\n",
       "      <td>Female</td>\n",
       "      <td>FIN</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>...</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>G80322</td>\n",
       "      <td>30478184</td>\n",
       "      <td>Plate 8</td>\n",
       "      <td>30.41078</td>\n",
       "      <td>34620</td>\n",
       "      <td>PDO-4157</td>\n",
       "      <td>G73882</td>\n",
       "      <td>28.400000</td>\n",
       "      <td>15 to 30 months</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>E002338</td>\n",
       "      <td>3107293</td>\n",
       "      <td>399</td>\n",
       "      <td>13</td>\n",
       "      <td>vaginal</td>\n",
       "      <td>285</td>\n",
       "      <td>Female</td>\n",
       "      <td>FIN</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>...</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>G80513</td>\n",
       "      <td>28503488</td>\n",
       "      <td>Plate 7</td>\n",
       "      <td>28.360762</td>\n",
       "      <td>7337</td>\n",
       "      <td>PDO-4356</td>\n",
       "      <td>G75849</td>\n",
       "      <td>13.300000</td>\n",
       "      <td>less than 15 months</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>E002338</td>\n",
       "      <td>3000144</td>\n",
       "      <td>58</td>\n",
       "      <td>2</td>\n",
       "      <td>vaginal</td>\n",
       "      <td>285</td>\n",
       "      <td>Female</td>\n",
       "      <td>FIN</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>...</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>G80490</td>\n",
       "      <td>18044614</td>\n",
       "      <td>Plate 7</td>\n",
       "      <td>17.982269</td>\n",
       "      <td>9209</td>\n",
       "      <td>PDO-4356</td>\n",
       "      <td>G75872</td>\n",
       "      <td>1.933333</td>\n",
       "      <td>less than 15 months</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 61 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Unnamed: 0 subjectID SampleID age_at_collection collection_month delivery  \\\n",
       "5            E002338  3101193               304               10  vaginal   \n",
       "6            E002338  3107294               493               16  vaginal   \n",
       "7            E002338  3113022               852               28  vaginal   \n",
       "8            E002338  3107293               399               13  vaginal   \n",
       "19           E002338  3000144                58                2  vaginal   \n",
       "\n",
       "Unnamed: 0 gest_time  gender country Exclusive_breast_feeding  \\\n",
       "5                285  Female     FIN                    FALSE   \n",
       "6                285  Female     FIN                    FALSE   \n",
       "7                285  Female     FIN                    FALSE   \n",
       "8                285  Female     FIN                    FALSE   \n",
       "19               285  Female     FIN                    FALSE   \n",
       "\n",
       "Unnamed: 0 Breast_feeding_end  ... allergy_timothy gid_wgs mgx_reads mgx_pool  \\\n",
       "5                       FALSE  ...           FALSE  G80541  28876314  Plate 7   \n",
       "6                        TRUE  ...           FALSE  G80537  29615868  Plate 7   \n",
       "7                        TRUE  ...           FALSE  G80322  30478184  Plate 8   \n",
       "8                       FALSE  ...           FALSE  G80513  28503488  Plate 7   \n",
       "19                      FALSE  ...           FALSE  G80490  18044614  Plate 7   \n",
       "\n",
       "Unnamed: 0 mgx_reads_filtered read_count_16S sequencing_PDO_16S gid_16s  \\\n",
       "5                   28.568962          15870           PDO-4356  G75694   \n",
       "6                   29.542272          13609           PDO-4356  G75855   \n",
       "7                    30.41078          34620           PDO-4157  G73882   \n",
       "8                   28.360762           7337           PDO-4356  G75849   \n",
       "19                  17.982269           9209           PDO-4356  G75872   \n",
       "\n",
       "Unnamed: 0 age_months            dev_stage  \n",
       "5           10.133333  less than 15 months  \n",
       "6           16.433333      15 to 30 months  \n",
       "7           28.400000      15 to 30 months  \n",
       "8           13.300000  less than 15 months  \n",
       "19           1.933333  less than 15 months  \n",
       "\n",
       "[5 rows x 61 columns]"
      ]
     },
     "execution_count": 106,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metadata_filtered.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>subjectID</th>\n",
       "      <th>SampleID</th>\n",
       "      <th>age_at_collection</th>\n",
       "      <th>collection_month</th>\n",
       "      <th>delivery</th>\n",
       "      <th>gest_time</th>\n",
       "      <th>gender</th>\n",
       "      <th>country</th>\n",
       "      <th>Exclusive_breast_feeding</th>\n",
       "      <th>Breast_feeding_end</th>\n",
       "      <th>...</th>\n",
       "      <th>allergy_birch</th>\n",
       "      <th>allergy_timothy</th>\n",
       "      <th>gid_wgs</th>\n",
       "      <th>mgx_reads</th>\n",
       "      <th>mgx_pool</th>\n",
       "      <th>mgx_reads_filtered</th>\n",
       "      <th>read_count_16S</th>\n",
       "      <th>sequencing_PDO_16S</th>\n",
       "      <th>gid_16s</th>\n",
       "      <th>age_months</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>dev_stage</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>15 to 30 months</th>\n",
       "      <td>340</td>\n",
       "      <td>340</td>\n",
       "      <td>340</td>\n",
       "      <td>340</td>\n",
       "      <td>334</td>\n",
       "      <td>334</td>\n",
       "      <td>340</td>\n",
       "      <td>340</td>\n",
       "      <td>340</td>\n",
       "      <td>340</td>\n",
       "      <td>...</td>\n",
       "      <td>314</td>\n",
       "      <td>314</td>\n",
       "      <td>340</td>\n",
       "      <td>340</td>\n",
       "      <td>340</td>\n",
       "      <td>340</td>\n",
       "      <td>340</td>\n",
       "      <td>340</td>\n",
       "      <td>340</td>\n",
       "      <td>340</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>less than 15 months</th>\n",
       "      <td>389</td>\n",
       "      <td>389</td>\n",
       "      <td>389</td>\n",
       "      <td>389</td>\n",
       "      <td>385</td>\n",
       "      <td>385</td>\n",
       "      <td>389</td>\n",
       "      <td>389</td>\n",
       "      <td>389</td>\n",
       "      <td>389</td>\n",
       "      <td>...</td>\n",
       "      <td>364</td>\n",
       "      <td>364</td>\n",
       "      <td>389</td>\n",
       "      <td>389</td>\n",
       "      <td>389</td>\n",
       "      <td>385</td>\n",
       "      <td>389</td>\n",
       "      <td>389</td>\n",
       "      <td>389</td>\n",
       "      <td>389</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>older than 30 months</th>\n",
       "      <td>46</td>\n",
       "      <td>46</td>\n",
       "      <td>46</td>\n",
       "      <td>46</td>\n",
       "      <td>45</td>\n",
       "      <td>45</td>\n",
       "      <td>46</td>\n",
       "      <td>46</td>\n",
       "      <td>46</td>\n",
       "      <td>46</td>\n",
       "      <td>...</td>\n",
       "      <td>44</td>\n",
       "      <td>44</td>\n",
       "      <td>46</td>\n",
       "      <td>46</td>\n",
       "      <td>46</td>\n",
       "      <td>46</td>\n",
       "      <td>46</td>\n",
       "      <td>46</td>\n",
       "      <td>46</td>\n",
       "      <td>46</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows × 60 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Unnamed: 0            subjectID  SampleID  age_at_collection  \\\n",
       "dev_stage                                                      \n",
       "15 to 30 months             340       340                340   \n",
       "less than 15 months         389       389                389   \n",
       "older than 30 months         46        46                 46   \n",
       "\n",
       "Unnamed: 0            collection_month  delivery  gest_time  gender  country  \\\n",
       "dev_stage                                                                      \n",
       "15 to 30 months                    340       334        334     340      340   \n",
       "less than 15 months                389       385        385     389      389   \n",
       "older than 30 months                46        45         45      46       46   \n",
       "\n",
       "Unnamed: 0            Exclusive_breast_feeding  Breast_feeding_end  ...  \\\n",
       "dev_stage                                                           ...   \n",
       "15 to 30 months                            340                 340  ...   \n",
       "less than 15 months                        389                 389  ...   \n",
       "older than 30 months                        46                  46  ...   \n",
       "\n",
       "Unnamed: 0            allergy_birch  allergy_timothy  gid_wgs  mgx_reads  \\\n",
       "dev_stage                                                                  \n",
       "15 to 30 months                 314              314      340        340   \n",
       "less than 15 months             364              364      389        389   \n",
       "older than 30 months             44               44       46         46   \n",
       "\n",
       "Unnamed: 0            mgx_pool  mgx_reads_filtered  read_count_16S  \\\n",
       "dev_stage                                                            \n",
       "15 to 30 months            340                 340             340   \n",
       "less than 15 months        389                 385             389   \n",
       "older than 30 months        46                  46              46   \n",
       "\n",
       "Unnamed: 0            sequencing_PDO_16S  gid_16s  age_months  \n",
       "dev_stage                                                      \n",
       "15 to 30 months                      340      340         340  \n",
       "less than 15 months                  389      389         389  \n",
       "older than 30 months                  46       46          46  \n",
       "\n",
       "[3 rows x 60 columns]"
      ]
     },
     "execution_count": 107,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metadata_filtered.groupby('dev_stage').count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [],
   "source": [
    "mgx = mgx_genus[children_intersect]\n",
    "amplicon = amp_genus[children_intersect]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/danielle/miniconda2/lib/python3.7/site-packages/ipykernel_launcher.py:2: DeprecationWarning: elementwise comparison failed; this will raise an error in the future.\n",
      "  \n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "False"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# confirm that we now have the same samples\n",
    "(amp_genus.columns.values) == (mgx_genus.columns.values)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [],
   "source": [
    "amplicon_melt = pd.melt(amplicon, id_vars=[\"taxa\"], var_name = \"sampleid\", value_name = \"amplicon_abund\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [],
   "source": [
    "mgx_melt = pd.melt(mgx, id_vars=[\"taxa\"], var_name = \"sampleid\", value_name = \"mgx_abund\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_taxa = pd.merge(amplicon_melt, mgx_melt, on = [\"sampleid\", \"taxa\"], how = \"outer\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_taxa.to_csv('taxa_abundance_comparison.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_taxa[\"abs_diff\"] = abs(merged_taxa[\"amplicon_abund\"] - merged_taxa[\"mgx_abund\"])\n",
    "merged_taxa[\"tot_diff\"] = (merged_taxa[\"amplicon_abund\"] - merged_taxa[\"mgx_abund\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>taxa</th>\n",
       "      <th>sampleid</th>\n",
       "      <th>amplicon_abund</th>\n",
       "      <th>mgx_abund</th>\n",
       "      <th>abs_diff</th>\n",
       "      <th>tot_diff</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>141285</th>\n",
       "      <td>Rothia</td>\n",
       "      <td>G78638</td>\n",
       "      <td>0.000195</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000195</td>\n",
       "      <td>0.000195</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>235159</th>\n",
       "      <td>Leclercia</td>\n",
       "      <td>G80309</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>166209</th>\n",
       "      <td>Enterorhabdus</td>\n",
       "      <td>G78511</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31158</th>\n",
       "      <td>Brevibacterium</td>\n",
       "      <td>G80267</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74615</th>\n",
       "      <td>Anaerocolumna</td>\n",
       "      <td>G78611</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>119096</th>\n",
       "      <td>Enterococcus</td>\n",
       "      <td>G78914</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>216400</th>\n",
       "      <td>Libanicoccus</td>\n",
       "      <td>G78796</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92270</th>\n",
       "      <td>Lactobacillus</td>\n",
       "      <td>G80591</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>257983</th>\n",
       "      <td>Bavariicoccus</td>\n",
       "      <td>G80296</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12003</th>\n",
       "      <td>Phascolarctobacterium</td>\n",
       "      <td>G80557</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                         taxa sampleid  amplicon_abund  mgx_abund  abs_diff  \\\n",
       "141285                 Rothia   G78638        0.000195        0.0  0.000195   \n",
       "235159              Leclercia   G80309             NaN        0.0       NaN   \n",
       "166209          Enterorhabdus   G78511        0.000000        0.0  0.000000   \n",
       "31158          Brevibacterium   G80267        0.000000        NaN       NaN   \n",
       "74615           Anaerocolumna   G78611        0.000000        0.0  0.000000   \n",
       "119096           Enterococcus   G78914        0.000000        0.0  0.000000   \n",
       "216400           Libanicoccus   G78796        0.000000        NaN       NaN   \n",
       "92270           Lactobacillus   G80591        0.000000        0.0  0.000000   \n",
       "257983          Bavariicoccus   G80296             NaN        0.0       NaN   \n",
       "12003   Phascolarctobacterium   G80557        0.000000        0.0  0.000000   \n",
       "\n",
       "        tot_diff  \n",
       "141285  0.000195  \n",
       "235159       NaN  \n",
       "166209  0.000000  \n",
       "31158        NaN  \n",
       "74615   0.000000  \n",
       "119096  0.000000  \n",
       "216400       NaN  \n",
       "92270   0.000000  \n",
       "257983       NaN  \n",
       "12003   0.000000  "
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_taxa.sample(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>taxa</th>\n",
       "      <th>sampleid</th>\n",
       "      <th>amplicon_abund</th>\n",
       "      <th>mgx_abund</th>\n",
       "      <th>abs_diff</th>\n",
       "      <th>tot_diff</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Abiotrophia</td>\n",
       "      <td>G69213</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Acetanaerobacterium</td>\n",
       "      <td>G69213</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Acetobacter</td>\n",
       "      <td>G69213</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Achromobacter</td>\n",
       "      <td>G69213</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Acidaminococcus</td>\n",
       "      <td>G69213</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Acinetobacter</td>\n",
       "      <td>G69213</td>\n",
       "      <td>0.000035</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000035</td>\n",
       "      <td>0.000035</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Actinobacillus</td>\n",
       "      <td>G69213</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Actinomyces</td>\n",
       "      <td>G69213</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Actinotignum</td>\n",
       "      <td>G69213</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Adlercreutzia</td>\n",
       "      <td>G69213</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000017</td>\n",
       "      <td>0.000017</td>\n",
       "      <td>-0.000017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Aerococcus</td>\n",
       "      <td>G69213</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Aeromonas</td>\n",
       "      <td>G69213</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Agathobacter</td>\n",
       "      <td>G69213</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Aggregatibacter</td>\n",
       "      <td>G69213</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Akkermansia</td>\n",
       "      <td>G69213</td>\n",
       "      <td>0.206776</td>\n",
       "      <td>0.468162</td>\n",
       "      <td>0.261386</td>\n",
       "      <td>-0.261386</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Alcaligenes</td>\n",
       "      <td>G69213</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Alcanivorax</td>\n",
       "      <td>G69213</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Alistipes</td>\n",
       "      <td>G69213</td>\n",
       "      <td>0.061982</td>\n",
       "      <td>0.081017</td>\n",
       "      <td>0.019035</td>\n",
       "      <td>-0.019035</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Allisonella</td>\n",
       "      <td>G69213</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Alloprevotella</td>\n",
       "      <td>G69213</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Alloscardovia</td>\n",
       "      <td>G69213</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>Anaerocolumna</td>\n",
       "      <td>G69213</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>Anaerofilum</td>\n",
       "      <td>G69213</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>Anaerofustis</td>\n",
       "      <td>G69213</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>Anaeroglobus</td>\n",
       "      <td>G69213</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>Anaeroplasma</td>\n",
       "      <td>G69213</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>Anaerosporobacter</td>\n",
       "      <td>G69213</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>Anaerostignum</td>\n",
       "      <td>G69213</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>Anaerostipes</td>\n",
       "      <td>G69213</td>\n",
       "      <td>0.020075</td>\n",
       "      <td>0.020533</td>\n",
       "      <td>0.000458</td>\n",
       "      <td>-0.000458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>Anaerotruncus</td>\n",
       "      <td>G69213</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>270445</th>\n",
       "      <td>Enterobacter</td>\n",
       "      <td>G78897</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>270446</th>\n",
       "      <td>Erysipelothrix</td>\n",
       "      <td>G78897</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>270447</th>\n",
       "      <td>Escherichia</td>\n",
       "      <td>G78897</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>270448</th>\n",
       "      <td>Finegoldia</td>\n",
       "      <td>G78897</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>270449</th>\n",
       "      <td>Franconibacter</td>\n",
       "      <td>G78897</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>270450</th>\n",
       "      <td>Gardnerella</td>\n",
       "      <td>G78897</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>270451</th>\n",
       "      <td>Gemmiger</td>\n",
       "      <td>G78897</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>270452</th>\n",
       "      <td>Hafnia</td>\n",
       "      <td>G78897</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>270453</th>\n",
       "      <td>Klebsiella</td>\n",
       "      <td>G78897</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>270454</th>\n",
       "      <td>Kluyvera</td>\n",
       "      <td>G78897</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>270455</th>\n",
       "      <td>Lawsonibacter</td>\n",
       "      <td>G78897</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>270456</th>\n",
       "      <td>Leclercia</td>\n",
       "      <td>G78897</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>270457</th>\n",
       "      <td>Massiliomicrobiota</td>\n",
       "      <td>G78897</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>270458</th>\n",
       "      <td>Metakosakonia</td>\n",
       "      <td>G78897</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>270459</th>\n",
       "      <td>Moraxella</td>\n",
       "      <td>G78897</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>270460</th>\n",
       "      <td>Oenococcus</td>\n",
       "      <td>G78897</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>270461</th>\n",
       "      <td>Pantoea</td>\n",
       "      <td>G78897</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>270462</th>\n",
       "      <td>Parvimonas</td>\n",
       "      <td>G78897</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>270463</th>\n",
       "      <td>Peptoniphilus</td>\n",
       "      <td>G78897</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>270464</th>\n",
       "      <td>Phytobacter</td>\n",
       "      <td>G78897</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>270465</th>\n",
       "      <td>Propionimicrobium</td>\n",
       "      <td>G78897</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>270466</th>\n",
       "      <td>Raoultella</td>\n",
       "      <td>G78897</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>270467</th>\n",
       "      <td>Ruthenibacterium</td>\n",
       "      <td>G78897</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.003036</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>270468</th>\n",
       "      <td>Saccharomyces</td>\n",
       "      <td>G78897</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>270469</th>\n",
       "      <td>Salmonella</td>\n",
       "      <td>G78897</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>270470</th>\n",
       "      <td>Serratia</td>\n",
       "      <td>G78897</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>270471</th>\n",
       "      <td>Sharpea</td>\n",
       "      <td>G78897</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>270472</th>\n",
       "      <td>Superficieibacter</td>\n",
       "      <td>G78897</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>270473</th>\n",
       "      <td>Turicimonas</td>\n",
       "      <td>G78897</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>270474</th>\n",
       "      <td>Yersinia</td>\n",
       "      <td>G78897</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>270475 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                       taxa sampleid  amplicon_abund  mgx_abund  abs_diff  \\\n",
       "0               Abiotrophia   G69213        0.000000   0.000000  0.000000   \n",
       "1       Acetanaerobacterium   G69213        0.000000   0.000000  0.000000   \n",
       "2               Acetobacter   G69213        0.000000   0.000000  0.000000   \n",
       "3             Achromobacter   G69213        0.000000   0.000000  0.000000   \n",
       "4           Acidaminococcus   G69213        0.000000   0.000000  0.000000   \n",
       "5             Acinetobacter   G69213        0.000035   0.000000  0.000035   \n",
       "6            Actinobacillus   G69213        0.000000   0.000000  0.000000   \n",
       "7               Actinomyces   G69213        0.000000   0.000000  0.000000   \n",
       "8              Actinotignum   G69213        0.000000   0.000000  0.000000   \n",
       "9             Adlercreutzia   G69213        0.000000   0.000017  0.000017   \n",
       "10               Aerococcus   G69213        0.000000   0.000000  0.000000   \n",
       "11                Aeromonas   G69213        0.000000   0.000000  0.000000   \n",
       "12             Agathobacter   G69213        0.000000   0.000000  0.000000   \n",
       "13          Aggregatibacter   G69213        0.000000   0.000000  0.000000   \n",
       "14              Akkermansia   G69213        0.206776   0.468162  0.261386   \n",
       "15              Alcaligenes   G69213        0.000000   0.000000  0.000000   \n",
       "16              Alcanivorax   G69213        0.000000   0.000000  0.000000   \n",
       "17                Alistipes   G69213        0.061982   0.081017  0.019035   \n",
       "18              Allisonella   G69213        0.000000   0.000000  0.000000   \n",
       "19           Alloprevotella   G69213        0.000000   0.000000  0.000000   \n",
       "20            Alloscardovia   G69213        0.000000   0.000000  0.000000   \n",
       "21            Anaerocolumna   G69213        0.000000   0.000000  0.000000   \n",
       "22              Anaerofilum   G69213        0.000000   0.000000  0.000000   \n",
       "23             Anaerofustis   G69213        0.000000   0.000000  0.000000   \n",
       "24             Anaeroglobus   G69213        0.000000   0.000000  0.000000   \n",
       "25             Anaeroplasma   G69213        0.000000   0.000000  0.000000   \n",
       "26        Anaerosporobacter   G69213        0.000000   0.000000  0.000000   \n",
       "27            Anaerostignum   G69213        0.000000   0.000000  0.000000   \n",
       "28             Anaerostipes   G69213        0.020075   0.020533  0.000458   \n",
       "29            Anaerotruncus   G69213        0.000000   0.000000  0.000000   \n",
       "...                     ...      ...             ...        ...       ...   \n",
       "270445         Enterobacter   G78897        0.000000   0.000000  0.000000   \n",
       "270446       Erysipelothrix   G78897        0.000000   0.000000  0.000000   \n",
       "270447          Escherichia   G78897        0.000000   0.000000  0.000000   \n",
       "270448           Finegoldia   G78897        0.000000   0.000000  0.000000   \n",
       "270449       Franconibacter   G78897        0.000000   0.000000  0.000000   \n",
       "270450          Gardnerella   G78897        0.000000   0.000000  0.000000   \n",
       "270451             Gemmiger   G78897        0.000000   0.000000  0.000000   \n",
       "270452               Hafnia   G78897        0.000000   0.000000  0.000000   \n",
       "270453           Klebsiella   G78897        0.000000   0.000000  0.000000   \n",
       "270454             Kluyvera   G78897        0.000000   0.000000  0.000000   \n",
       "270455        Lawsonibacter   G78897        0.000000   0.000000  0.000000   \n",
       "270456            Leclercia   G78897        0.000000   0.000000  0.000000   \n",
       "270457   Massiliomicrobiota   G78897        0.000000   0.000000  0.000000   \n",
       "270458        Metakosakonia   G78897        0.000000   0.000000  0.000000   \n",
       "270459            Moraxella   G78897        0.000000   0.000000  0.000000   \n",
       "270460           Oenococcus   G78897        0.000000   0.000000  0.000000   \n",
       "270461              Pantoea   G78897        0.000000   0.000000  0.000000   \n",
       "270462           Parvimonas   G78897        0.000000   0.000000  0.000000   \n",
       "270463        Peptoniphilus   G78897        0.000000   0.000000  0.000000   \n",
       "270464          Phytobacter   G78897        0.000000   0.000000  0.000000   \n",
       "270465    Propionimicrobium   G78897        0.000000   0.000000  0.000000   \n",
       "270466           Raoultella   G78897        0.000000   0.000000  0.000000   \n",
       "270467     Ruthenibacterium   G78897        0.000000   0.003036  0.000000   \n",
       "270468        Saccharomyces   G78897        0.000000   0.000000  0.000000   \n",
       "270469           Salmonella   G78897        0.000000   0.000000  0.000000   \n",
       "270470             Serratia   G78897        0.000000   0.000000  0.000000   \n",
       "270471              Sharpea   G78897        0.000000   0.000000  0.000000   \n",
       "270472    Superficieibacter   G78897        0.000000   0.000000  0.000000   \n",
       "270473          Turicimonas   G78897        0.000000   0.000000  0.000000   \n",
       "270474             Yersinia   G78897        0.000000   0.000000  0.000000   \n",
       "\n",
       "        tot_diff  \n",
       "0       0.000000  \n",
       "1       0.000000  \n",
       "2       0.000000  \n",
       "3       0.000000  \n",
       "4       0.000000  \n",
       "5       0.000035  \n",
       "6       0.000000  \n",
       "7       0.000000  \n",
       "8       0.000000  \n",
       "9      -0.000017  \n",
       "10      0.000000  \n",
       "11      0.000000  \n",
       "12      0.000000  \n",
       "13      0.000000  \n",
       "14     -0.261386  \n",
       "15      0.000000  \n",
       "16      0.000000  \n",
       "17     -0.019035  \n",
       "18      0.000000  \n",
       "19      0.000000  \n",
       "20      0.000000  \n",
       "21      0.000000  \n",
       "22      0.000000  \n",
       "23      0.000000  \n",
       "24      0.000000  \n",
       "25      0.000000  \n",
       "26      0.000000  \n",
       "27      0.000000  \n",
       "28     -0.000458  \n",
       "29      0.000000  \n",
       "...          ...  \n",
       "270445  0.000000  \n",
       "270446  0.000000  \n",
       "270447  0.000000  \n",
       "270448  0.000000  \n",
       "270449  0.000000  \n",
       "270450  0.000000  \n",
       "270451  0.000000  \n",
       "270452  0.000000  \n",
       "270453  0.000000  \n",
       "270454  0.000000  \n",
       "270455  0.000000  \n",
       "270456  0.000000  \n",
       "270457  0.000000  \n",
       "270458  0.000000  \n",
       "270459  0.000000  \n",
       "270460  0.000000  \n",
       "270461  0.000000  \n",
       "270462  0.000000  \n",
       "270463  0.000000  \n",
       "270464  0.000000  \n",
       "270465  0.000000  \n",
       "270466  0.000000  \n",
       "270467  0.000000  \n",
       "270468  0.000000  \n",
       "270469  0.000000  \n",
       "270470  0.000000  \n",
       "270471  0.000000  \n",
       "270472  0.000000  \n",
       "270473  0.000000  \n",
       "270474  0.000000  \n",
       "\n",
       "[270475 rows x 6 columns]"
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_taxa.fillna(0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [],
   "source": [
    "amplicon_avg_abund = merged_taxa.groupby(\"taxa\")[\"amplicon_abund\"].mean()\n",
    "mgx_avg_abund = merged_taxa.groupby(\"taxa\")[\"mgx_abund\"].mean()\n",
    "taxa_list = sorted(set(merged_taxa[\"taxa\"]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [],
   "source": [
    "mean_taxa_abund = pd.DataFrame(\n",
    "    (zip(taxa_list, amplicon_avg_abund, mgx_avg_abund)),  \n",
    "    columns = ['taxa','amp_avg_abund', 'mgx_avg_abund'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [],
   "source": [
    "mean_taxa_abund[\"abs_diff\"] = abs(mean_taxa_abund[\"amp_avg_abund\"] - mean_taxa_abund[\"mgx_avg_abund\"])\n",
    "mean_taxa_abund[\"total_diff\"] = mean_taxa_abund[\"amp_avg_abund\"] - mean_taxa_abund[\"mgx_avg_abund\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [],
   "source": [
    "mean_taxa_abund.sort_values(\"abs_diff\", axis = 0, ascending = True, \n",
    "                 inplace = True, na_position ='last')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>taxa</th>\n",
       "      <th>amp_avg_abund</th>\n",
       "      <th>mgx_avg_abund</th>\n",
       "      <th>abs_diff</th>\n",
       "      <th>total_diff</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>146</th>\n",
       "      <td>Fretibacterium</td>\n",
       "      <td>3.600000e-08</td>\n",
       "      <td>2.458065e-07</td>\n",
       "      <td>2.098065e-07</td>\n",
       "      <td>-2.098065e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>194</th>\n",
       "      <td>Legionella</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>131</th>\n",
       "      <td>Erysipelothrix</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.566645e-05</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>284</th>\n",
       "      <td>Rhodococcus</td>\n",
       "      <td>1.481290e-07</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>105</th>\n",
       "      <td>Dechloromonas</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>120</th>\n",
       "      <td>Eikenella</td>\n",
       "      <td>3.213077e-06</td>\n",
       "      <td>4.806452e-07</td>\n",
       "      <td>2.732432e-06</td>\n",
       "      <td>2.732432e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>Allisonella</td>\n",
       "      <td>1.130362e-04</td>\n",
       "      <td>2.307484e-05</td>\n",
       "      <td>8.996133e-05</td>\n",
       "      <td>8.996133e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208</th>\n",
       "      <td>Mesorhizobium</td>\n",
       "      <td>1.038710e-07</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>274</th>\n",
       "      <td>Psychrobacillus</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>Buttiauxella</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5.984516e-07</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                taxa  amp_avg_abund  mgx_avg_abund      abs_diff    total_diff\n",
       "146   Fretibacterium   3.600000e-08   2.458065e-07  2.098065e-07 -2.098065e-07\n",
       "194       Legionella   0.000000e+00            NaN           NaN           NaN\n",
       "131   Erysipelothrix            NaN   1.566645e-05           NaN           NaN\n",
       "284      Rhodococcus   1.481290e-07            NaN           NaN           NaN\n",
       "105    Dechloromonas   0.000000e+00            NaN           NaN           NaN\n",
       "120        Eikenella   3.213077e-06   4.806452e-07  2.732432e-06  2.732432e-06\n",
       "23       Allisonella   1.130362e-04   2.307484e-05  8.996133e-05  8.996133e-05\n",
       "208    Mesorhizobium   1.038710e-07            NaN           NaN           NaN\n",
       "274  Psychrobacillus   0.000000e+00            NaN           NaN           NaN\n",
       "67      Buttiauxella            NaN   5.984516e-07           NaN           NaN"
      ]
     },
     "execution_count": 76,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mean_taxa_abund.sample(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [],
   "source": [
    "mean_taxa_abund.to_csv('taxa_difference.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## making giant dataframe of abundances"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [],
   "source": [
    "amp_trans = amplicon.set_index(\"taxa\").transpose()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [],
   "source": [
    "amp_trans.reset_index(level=0, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [],
   "source": [
    "amp_trans.rename(columns = {'index':'sampleid'}, inplace = True) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [],
   "source": [
    "amp_trans[\"uid\"] = amp_trans[\"sampleid\"].astype(str)+'-amp'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [],
   "source": [
    "amp_trans[\"method\"] = \"amp\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>taxa</th>\n",
       "      <th>sampleid</th>\n",
       "      <th>Abiotrophia</th>\n",
       "      <th>Acetanaerobacterium</th>\n",
       "      <th>Acetobacter</th>\n",
       "      <th>Achromobacter</th>\n",
       "      <th>Acidaminococcus</th>\n",
       "      <th>Acinetobacter</th>\n",
       "      <th>Actinobacillus</th>\n",
       "      <th>Actinomyces</th>\n",
       "      <th>Actinotignum</th>\n",
       "      <th>...</th>\n",
       "      <th>Veillonellaceae</th>\n",
       "      <th>Verticiella</th>\n",
       "      <th>Victivallaceae</th>\n",
       "      <th>Victivallis</th>\n",
       "      <th>Vulcaniibacterium</th>\n",
       "      <th>Weissella</th>\n",
       "      <th>Xanthomonas</th>\n",
       "      <th>Zoogloea</th>\n",
       "      <th>uid</th>\n",
       "      <th>method</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>G69213</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000035</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>G69213-amp</td>\n",
       "      <td>amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>G80594</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>G80594-amp</td>\n",
       "      <td>amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>G78901</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000338</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000879</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>G78901-amp</td>\n",
       "      <td>amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>G80543</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>G80543-amp</td>\n",
       "      <td>amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>G80417</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.001079</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>G80417-amp</td>\n",
       "      <td>amp</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 305 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "taxa sampleid  Abiotrophia  Acetanaerobacterium  Acetobacter  Achromobacter  \\\n",
       "0      G69213          0.0                  0.0          0.0            0.0   \n",
       "1      G80594          0.0                  0.0          0.0            0.0   \n",
       "2      G78901          0.0                  0.0          0.0            0.0   \n",
       "3      G80543          0.0                  0.0          0.0            0.0   \n",
       "4      G80417          0.0                  0.0          0.0            0.0   \n",
       "\n",
       "taxa  Acidaminococcus  Acinetobacter  Actinobacillus  Actinomyces  \\\n",
       "0                 0.0       0.000035             0.0     0.000000   \n",
       "1                 0.0       0.000000             0.0     0.000000   \n",
       "2                 0.0       0.000338             0.0     0.000879   \n",
       "3                 0.0       0.000000             0.0     0.000000   \n",
       "4                 0.0       0.000000             0.0     0.001079   \n",
       "\n",
       "taxa  Actinotignum  ...  Veillonellaceae  Verticiella  Victivallaceae  \\\n",
       "0              0.0  ...              0.0          0.0             0.0   \n",
       "1              0.0  ...              0.0          0.0             0.0   \n",
       "2              0.0  ...              0.0          0.0             0.0   \n",
       "3              0.0  ...              0.0          0.0             0.0   \n",
       "4              0.0  ...              0.0          0.0             0.0   \n",
       "\n",
       "taxa  Victivallis  Vulcaniibacterium  Weissella  Xanthomonas  Zoogloea  \\\n",
       "0             0.0                0.0        0.0          0.0       0.0   \n",
       "1             0.0                0.0        0.0          0.0       0.0   \n",
       "2             0.0                0.0        0.0          0.0       0.0   \n",
       "3             0.0                0.0        0.0          0.0       0.0   \n",
       "4             0.0                0.0        0.0          0.0       0.0   \n",
       "\n",
       "taxa         uid  method  \n",
       "0     G69213-amp     amp  \n",
       "1     G80594-amp     amp  \n",
       "2     G78901-amp     amp  \n",
       "3     G80543-amp     amp  \n",
       "4     G80417-amp     amp  \n",
       "\n",
       "[5 rows x 305 columns]"
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "amp_trans.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [],
   "source": [
    "mgx_trans = mgx.set_index(\"taxa\").transpose()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [],
   "source": [
    "mgx_trans.reset_index(level=0, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [],
   "source": [
    "mgx_trans.rename(columns = {'index':'sampleid'}, inplace = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [],
   "source": [
    "mgx_trans[\"uid\"] = mgx_trans[\"sampleid\"].astype(str)+'-mgx'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [],
   "source": [
    "mgx_trans[\"method\"] = \"mgx\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>taxa</th>\n",
       "      <th>sampleid</th>\n",
       "      <th>Absiella</th>\n",
       "      <th>Acidaminococcus</th>\n",
       "      <th>Acinetobacter</th>\n",
       "      <th>Actinobaculum</th>\n",
       "      <th>Actinomyces</th>\n",
       "      <th>Actinotignum</th>\n",
       "      <th>Adlercreutzia</th>\n",
       "      <th>Aeriscardovia</th>\n",
       "      <th>Aeromonas</th>\n",
       "      <th>...</th>\n",
       "      <th>Turicimonas</th>\n",
       "      <th>Tyzzerella</th>\n",
       "      <th>UNCLASSIFIED</th>\n",
       "      <th>Varibaculum</th>\n",
       "      <th>Veillonella</th>\n",
       "      <th>Victivallis</th>\n",
       "      <th>Weissella</th>\n",
       "      <th>Yersinia</th>\n",
       "      <th>uid</th>\n",
       "      <th>method</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>G69213</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000017</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.012562</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>G69213-mgx</td>\n",
       "      <td>mgx</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>G80594</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000101</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.010218</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>G80594-mgx</td>\n",
       "      <td>mgx</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>G78901</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000284</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000270</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000364</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>G78901-mgx</td>\n",
       "      <td>mgx</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>G80543</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.001198</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.007696</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>G80543-mgx</td>\n",
       "      <td>mgx</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>G80417</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000098</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>G80417-mgx</td>\n",
       "      <td>mgx</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 201 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "taxa sampleid  Absiella  Acidaminococcus  Acinetobacter  Actinobaculum  \\\n",
       "0      G69213       0.0              0.0       0.000000            0.0   \n",
       "1      G80594       0.0              0.0       0.000000            0.0   \n",
       "2      G78901       0.0              0.0       0.000000            0.0   \n",
       "3      G80543       0.0              0.0       0.001198            0.0   \n",
       "4      G80417       0.0              0.0       0.000000            0.0   \n",
       "\n",
       "taxa  Actinomyces  Actinotignum  Adlercreutzia  Aeriscardovia  Aeromonas  ...  \\\n",
       "0        0.000000           0.0       0.000017       0.000000        0.0  ...   \n",
       "1        0.000000           0.0       0.000000       0.000101        0.0  ...   \n",
       "2        0.000284           0.0       0.000000       0.000270        0.0  ...   \n",
       "3        0.000000           0.0       0.000000       0.000000        0.0  ...   \n",
       "4        0.000000           0.0       0.000000       0.000098        0.0  ...   \n",
       "\n",
       "taxa  Turicimonas  Tyzzerella  UNCLASSIFIED  Varibaculum  Veillonella  \\\n",
       "0             0.0         0.0      0.012562          0.0     0.000000   \n",
       "1             0.0         0.0      0.000000          0.0     0.010218   \n",
       "2             0.0         0.0      0.000000          0.0     0.000364   \n",
       "3             0.0         0.0      0.000000          0.0     0.007696   \n",
       "4             0.0         0.0      0.000000          0.0     0.000000   \n",
       "\n",
       "taxa  Victivallis  Weissella  Yersinia         uid  method  \n",
       "0             0.0        0.0       0.0  G69213-mgx     mgx  \n",
       "1             0.0        0.0       0.0  G80594-mgx     mgx  \n",
       "2             0.0        0.0       0.0  G78901-mgx     mgx  \n",
       "3             0.0        0.0       0.0  G80543-mgx     mgx  \n",
       "4             0.0        0.0       0.0  G80417-mgx     mgx  \n",
       "\n",
       "[5 rows x 201 columns]"
      ]
     },
     "execution_count": 89,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mgx_trans.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [],
   "source": [
    "concat_df = pd.concat([mgx_trans,amp_trans], sort=True).reset_index(drop = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1550"
      ]
     },
     "execution_count": 91,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(concat_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [
    {
     "ename": "FileNotFoundError",
     "evalue": "[Errno 2] File b'/Users/danielle/Documents/thesis/analysis/metadatawide.csv' does not exist: b'/Users/danielle/Documents/thesis/analysis/metadatawide.csv'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-92-7446d043851e>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mage\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread_csv\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"~/Documents/thesis/analysis/metadatawide.csv\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m~/miniconda2/lib/python3.7/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36mparser_f\u001b[0;34m(filepath_or_buffer, sep, delimiter, header, names, index_col, usecols, squeeze, prefix, mangle_dupe_cols, dtype, engine, converters, true_values, false_values, skipinitialspace, skiprows, skipfooter, nrows, na_values, keep_default_na, na_filter, verbose, skip_blank_lines, parse_dates, infer_datetime_format, keep_date_col, date_parser, dayfirst, iterator, chunksize, compression, thousands, decimal, lineterminator, quotechar, quoting, doublequote, escapechar, comment, encoding, dialect, tupleize_cols, error_bad_lines, warn_bad_lines, delim_whitespace, low_memory, memory_map, float_precision)\u001b[0m\n\u001b[1;32m    700\u001b[0m                     skip_blank_lines=skip_blank_lines)\n\u001b[1;32m    701\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 702\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0m_read\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfilepath_or_buffer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkwds\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    703\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    704\u001b[0m     \u001b[0mparser_f\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__name__\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mname\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda2/lib/python3.7/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36m_read\u001b[0;34m(filepath_or_buffer, kwds)\u001b[0m\n\u001b[1;32m    427\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    428\u001b[0m     \u001b[0;31m# Create the parser.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 429\u001b[0;31m     \u001b[0mparser\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mTextFileReader\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfilepath_or_buffer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwds\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    430\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    431\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0mchunksize\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0miterator\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda2/lib/python3.7/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, f, engine, **kwds)\u001b[0m\n\u001b[1;32m    893\u001b[0m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0moptions\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'has_index_names'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mkwds\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'has_index_names'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    894\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 895\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_make_engine\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mengine\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    896\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    897\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mclose\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda2/lib/python3.7/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36m_make_engine\u001b[0;34m(self, engine)\u001b[0m\n\u001b[1;32m   1120\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_make_engine\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mengine\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'c'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1121\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mengine\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m'c'\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1122\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_engine\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mCParserWrapper\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mf\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0moptions\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1123\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1124\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mengine\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m'python'\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda2/lib/python3.7/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, src, **kwds)\u001b[0m\n\u001b[1;32m   1851\u001b[0m         \u001b[0mkwds\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'usecols'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0musecols\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1852\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1853\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_reader\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mparsers\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mTextReader\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msrc\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwds\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1854\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0munnamed_cols\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_reader\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0munnamed_cols\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1855\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/parsers.pyx\u001b[0m in \u001b[0;36mpandas._libs.parsers.TextReader.__cinit__\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/parsers.pyx\u001b[0m in \u001b[0;36mpandas._libs.parsers.TextReader._setup_parser_source\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;31mFileNotFoundError\u001b[0m: [Errno 2] File b'/Users/danielle/Documents/thesis/analysis/metadatawide.csv' does not exist: b'/Users/danielle/Documents/thesis/analysis/metadatawide.csv'"
     ]
    }
   ],
   "source": [
    "age = pd.read_csv(\"~/Documents/thesis/analysis/metadatawide.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "age = age[['sample','childAgeMonths']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "age[\"sample\"] = age[\"sample\"].str.replace(\"_\",'-')\n",
    "age.rename(columns = {'sample':'sampleid'}, inplace = True) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# make age dictionary\n",
    "agedict = {str(s): {} for s in age[\"sampleid\"]}\n",
    "for index, row in age.iterrows():\n",
    "    age_months = row[\"childAgeMonths\"]\n",
    "    agedict[row[\"sampleid\"]]= age_months"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "concat_df[\"AgeMonths\"]= concat_df[\"sampleid\"].map(agedict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "cols_to_order = ['uid', 'sampleid',\"method\", \"AgeMonths\"]\n",
    "new_columns = cols_to_order + (concat_df.columns.drop(cols_to_order).tolist())\n",
    "concat_df = concat_df[new_columns]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "concat_df.sample(15)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "concat_df.to_csv('transposed_mgxamp_df.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "pwd()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
